<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prepare input data for peakCombiner package — prepare_input_regions • peakCombiner</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare input data for peakCombiner package — prepare_input_regions"><meta property="og:description" content="prepare_input_regions prepares the input data in the format
needed for all of the following steps within peakCombiner. It accepts the
following formats:
in memory data frame listing each sample's peak file location,
in memory data frame listing the peaks themselves that are found in each
sample, or
in memory GRanges object listing the peaks themselves that are found in
each sample.

"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">peakCombiner</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/peakCombiner.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/novartis/peakCombiner/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare input data for peakCombiner package</h1>
    <small class="dont-index">Source: <a href="https://github.com/novartis/peakCombiner/blob/devel/R/prepare_input_regions.R" class="external-link"><code>R/prepare_input_regions.R</code></a></small>
    <div class="hidden name"><code>prepare_input_regions.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>prepare_input_regions prepares the input data in the format
needed for all of the following steps within peakCombiner. It accepts the
following formats:</p><ul><li><p>in memory data frame listing each sample's peak file location,</p></li>
<li><p>in memory data frame listing the peaks themselves that are found in each
sample, or</p></li>
<li><p>in memory GRanges object listing the peaks themselves that are found in
each sample.</p></li>
</ul></div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prepare_input_regions</span><span class="op">(</span><span class="va">data</span>, show_messages <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>Data frame or GRanges object with the input data. Several
formats are accepted, which are described in full in the Details below.</p><ul><li><p>in memory data frame listing each sample's peak file location,</p></li>
<li><p>in memory data frame listing the peaks themselves that are found in each
sample, or</p></li>
<li><p>in memory GRanges object listing the peaks themselves that are found in
each sample.</p></li>
</ul></dd>


<dt id="arg-show-messages">show_messages<a class="anchor" aria-label="anchor" href="#arg-show-messages"></a></dt>
<dd><p>Logical value of TRUE (default) or FALSE. Defines if
info messages are displayed or not.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A tibble with the columns <code>chrom</code>, <code>start</code>, <code>end</code>, <code>name</code>, <code>score</code>,
<code>strand</code>, <code>center</code>, <code>sample_name</code>. The definitions of these columns are
described in full in the Details below. Use as input for functions
<code><a href="center_expand_regions.html">center_expand_regions()</a></code>, <code><a href="filter_regions.html">filter_regions()</a></code> and
<code><a href="combine_regions.html">combine_regions()</a></code>.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Accepted inputs are one of the three following options:</p><ol><li><p>In memory data frame listing each sample's peak file location</p><ul><li><p><code>sample_name</code> -  Unique name for each sample
(required).</p></li>
<li><p><code>path_peaks</code> -   Path to the file in which the genomic regions are
stored. For example, the path to a bed file or
<code>.narrowPeak</code> file (required).</p></li>
<li><p><code>file_format</code> -  The expected file format. Needed to correctly label the
columns of the input. Acceptable values are:
<code>bed</code>, <code>narrowPeak</code>, and <code>broadPeak</code> (required).</p></li>
<li><p><code>score_column</code> - The number of the column having the
the metric used to rank peak importance, where bigger
values are more important. If not provided,
column 9 will be used for <code>.narrowPeak</code> or
<code>.broadPeak</code> file formats. Column 9 corresponds to
the <code>qValue</code> as described in the UCSC documentation
<a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12" class="external-link">here</a>.
Other alternatives for <code>narrowPeak</code> or <code>broadPeak</code>
could be columns 7 or 8, which correspond to
<code>signalValue</code> or <code>pValue</code> (optional).</p></li>
</ul></li>
<li><p>In memory data frame listing the peaks themselves that are found in each
sample. The columns can be provided in any order and have the following
names. Note that additional columns will be dropped.</p><ul><li><p><code>chrom</code> - chromosome name (required).</p></li>
<li><p><code>start</code> - start coordinate of range (1-based coordinate system,
NOT like bed files which are 0-based) (required).</p></li>
<li><p><code>end</code> -   end coordinate of range (required).</p></li>
<li><p><code>sample_name</code> - unique identifier for a sample. No restrictions on
characters (required).</p></li>
<li><p><code>score</code> - the metric used to rank peak importance, where bigger values
are more important. For example, qValue from Macs2,
-log10FDR from another method, or fold enrichment over
background computed from your favorite method. If not
provided, defaults to 0 (optional).</p></li>
<li><p><code>strand</code> - values are '+', '-', or '.'. If not provided, defaults to '.'
(optional).</p></li>
<li><p><code>summit</code> - distance of the strongest signal ("summit") of the peak
region from the start coordinate (optional).</p></li>
</ul></li>
<li><p>In memory GRanges object listing the peaks themselves that are found in
each sample. This object is very similar to the data frame above,
except that <code>chrom</code>, <code>start</code>, and <code>end</code> are instead described using
the <code>GRanges</code> nomenclature. Note that additional columns will be dropped.</p></li>
</ol><p>This function parses the inputs provided and returns a data frame having the
columns listed below.</p><ul><li><p><code>chrom</code> -        chromosome name</p></li>
<li><p><code>start</code> -        start coordinate of range (1-based coordinate system,
NOT like bed files which are 0-based)</p></li>
<li><p><code>end</code> -          end coordinate of range</p></li>
<li><p><code>name</code> -         unique identifier for a region. auto-generated by this
function</p></li>
<li><p><code>score</code> -        the metric used to rank peak importance, where bigger
values are more important. For example, qValue from MACS2,
-log10FDR from another method, or fold enrichment over
background computed from your favorite method</p></li>
<li><p><code>strand</code> -       values are '+', '-', or '.'. Chromatin data are typically
non- stranded so will have a '.'.</p></li>
<li><p><code>center</code> -       absolute genomic coordinate of the nucleotide at the
center of the peak region, or alternatively the strongest
signal ("summit") of the peak region. If no value is
provided by the user, <code>center</code> defaults to the arithmetic
center of the peak region.</p></li>
<li><p><code>sample_name</code> -  unique identifier for a sample. No restrictions on
characters</p></li>
</ul><p>In addition, input data is checked for mutliple entries of the same genomic
region. This can occure when using called peak files as multiple summits can
be annotated within the sqme genomic regions (defined by <code>chrom</code>, <code>start</code>
and <code>end</code>). To avoid mutliple entries, this script is checking the input for
multiple summits within the same regions and maintains only the strongest
enriched (based on the values in the column <code>score</code>). This step is mantory
to quaranty an optimal result.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load in and prepare a an accepted tibble</span></span></span>
<span class="r-in"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">syn_data_tibble</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">data_prepared</span> <span class="op">&lt;-</span> <span class="fu">prepare_input_regions</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">syn_data_tibble</span>,</span></span>
<span class="r-in"><span>  show_messages <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Provide input `data` is a pre-loaded <span style="color: #0000BB;">&lt;data.frame&gt;</span> with the required column</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   names.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Start preparing data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Required columns will be added if missing.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Column <span style="color: #00BB00;">'name'</span> from `data` will be overwritten.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Column <span style="color: #00BB00;">'name'</span> is a computed column from <span style="color: #0000BB;">peakCombiner</span> and therefore</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   pre-exisiting data in a column <span style="color: #00BB00;">'name'</span> will not be retained.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Column <span style="color: #00BB00;">'center'</span> does exist in `data_prepared`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → The column <span style="color: #00BB00;">'center'</span> is taken to define output column <span style="color: #00BB00;">'center'</span>.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Checking <span style="color: #0000BB;">&lt;class&gt;</span> and <span style="color: #0000BB;">"values"</span> of all columns.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Checking whether duplicated regions exist and need to be collapsed.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> → Checked whether duplicated regions exist and need to be collapsed.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Duplicated regions identified and collapsed to unique <span style="color: #00BB00;">chrom</span>, <span style="color: #00BB00;">start</span>, and <span style="color: #00BB00;">end</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   for each sample by strongest <span style="color: #00BB00;">score</span> value.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Preparation of data finished successfully.</span>
<span class="r-in"><span><span class="va">data_prepared</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 52 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    chrom  start   end name            score strand center sample_name </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control_rep1|9     80 .         700 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control_rep1|1     96 .         600 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control_rep1|7     75 .         600 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control_rep1|8     50 .         600 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control_rep1|10    30 .         600 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control_rep1|11    20 .         600 control_rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 42 more rows</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Or a pre-loaded tibble with genomic regions and named columns.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">syn_data_control01</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">syn_data_treatment01</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">combined_input</span> <span class="op">&lt;-</span> <span class="va">syn_data_control01</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"control-rep1"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">syn_data_treatment01</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"treatment-rep1"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">prepare_input_regions</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">combined_input</span>,</span></span>
<span class="r-in"><span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 19 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    chrom  start   end name              score strand center sample_name   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control-rep1|9       80 .         700 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control-rep1|1       96 .         600 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control-rep1|2       95 .        <span style="text-decoration: underline;">3</span>800 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control-rep1|3       46 .        <span style="text-decoration: underline;">4</span>900 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control-rep1|5       26 .        <span style="text-decoration: underline;">5</span>700 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control-rep1|6       25 .        <span style="text-decoration: underline;">7</span>000 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control-rep1|7       75 .         600 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control-rep1|8       50 .         600 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control-rep1|10      30 .         600 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control-rep1|11      20 .         600 control-rep1  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> chr1     200   900 treatment-rep1|1    100 .         500 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> chr1     201   900 treatment-rep1|2    100 .         500 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200 treatment-rep1|3     99 .        <span style="text-decoration: underline;">2</span>800 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 treatment-rep1|4     60 .        <span style="text-decoration: underline;">4</span>700 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300 treatment-rep1|6     30 .        <span style="text-decoration: underline;">6</span>100 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 treatment-rep1|7     29 .        <span style="text-decoration: underline;">7</span>000 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span> chr10    201   900 treatment-rep1|8     95 .         500 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">18</span> chr2     201   900 treatment-rep1|9     50 .         500 treatment-rep1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">19</span> chr4-2   201   900 treatment-rep1|10    30 .         500 treatment-rep1</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Markus Muckenhuber, Charlotte Soneson, Michael Stadler, Kathleen Sprouffske.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

      </footer></div>






  </body></html>

