<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>peakCombiner: The R package to curate and merge enriched genomic regions into consensus peak sets • peakCombiner</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="peakCombiner: The R package to curate and merge enriched genomic regions into consensus peak sets">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">peakCombiner</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/peakCombiner.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/novartis/peakCombiner/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>peakCombiner: The R package to curate and merge
enriched genomic regions into consensus peak sets</h1>
                        <h4 data-toc-skip class="author">Markus
Muckenhuber, Michael Stadler, Kathleen Sprouffske</h4>
            
            <h4 data-toc-skip class="date">2025-02-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/novartis/peakCombiner/blob/devel/vignettes/peakCombiner.Rmd" class="external-link"><code>vignettes/peakCombiner.Rmd</code></a></small>
      <div class="hidden name"><code>peakCombiner.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette for the R package <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
guides you through the preparation of all accepted input files and how
to define the best parameters for your analysis. Our goal is to provide
you with an easy to understand, transparent and modular framework to
create the consensus regions file you need to address your scientific
question.</p>
<div class="section level3">
<h3 id="abstract">Abstract<a class="anchor" aria-label="anchor" href="#abstract"></a>
</h3>
<p>Genome-wide epigenomic data sets like ChIP-seq and ATAC-seq typically
use tools (e.g. MACS <span class="citation">(Zhang et al. 2008)</span>)
to identify genomic regions of interest, so-called peaks, usually for
multiple sample replicates and across experimental conditions. Many
downstream analyses require a consensus set of genomic regions relevant
to the experiment, but current tools within the R ecosystem to easily
and flexibly create combined peak sets from conditions and replicates
are limited. Here, we describe <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>,
a fully R-based, user-friendly, transparent, and flexible tool that
allows even novice R users to create a high-quality consensus peak list.
The modularity of its functions allows an easy way to filter and
recenter input and output data. A broad range of accepted input data
formats can be used as input to <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>,
and the resulting consensus regions set can be exported to a file or
used immediately as the starting point for most downstream peak
analyses.</p>
</div>
<div class="section level3">
<h3 id="input-file-formats">Input file formats<a class="anchor" aria-label="anchor" href="#input-file-formats"></a>
</h3>
</div>
<div class="section level3">
<h3 id="overview-of-steps">Overview of steps<a class="anchor" aria-label="anchor" href="#overview-of-steps"></a>
</h3>
<p>The package <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
contains four main functions to curate and combine genomic regions into
one set of consensus regions. A short overview about each the functions
is in the table below.</p>
<table class="table">
<colgroup>
<col width="32%">
<col width="30%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">Function name</th>
<th align="left">What it does</th>
<th align="left">Where to learn more</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><code>prepare_input_regions</code></td>
<td align="left">Transforms your data into the format needed by <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
for all following steps.</td>
<td align="left">See in section <a href="#prepare-input-data">Prepare
input data</a>.</td>
</tr>
<tr class="even">
<td align="left"><a href="#center-and-expand-regions"><code>center_expand_regions</code></a></td>
<td align="left">Modifies your genomic regions by centering and then
expanding them to a uniform size.</td>
<td align="left">See in section <a href="#center-and-expand-regions">Center and expand regions</a>.</td>
</tr>
<tr class="odd">
<td align="left"><a href="#filter-regions"><code>filter_regions</code></a></td>
<td align="left">Allows you to filter your genomic regions.</td>
<td align="left">See in section <a href="#filter-regions">Filter
regions</a>.</td>
</tr>
<tr class="even">
<td align="left"><a href="#combine-regions"><code>combine_regions</code></a></td>
<td align="left">Combine overlapping genomic regions to create a single
set of consensus genomic regions.</td>
<td align="left">See in section <a href="#combine-regions">Combine
regions</a>.</td>
</tr>
</tbody>
</table>
<p>Each <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
function has the parameter <code>show_messages</code> that by default
prints feedback messages as functions run. If you plan to use <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
non-interactively in a workflow, you can silence these messages by
setting <code>show_messages = FALSE</code>. Note that error messages
will still be printed to help you to troubleshoot potential
problems.</p>
</div>
<div class="section level3">
<h3 id="standard-genomic-regions-format">Standard genomic regions format<a class="anchor" aria-label="anchor" href="#standard-genomic-regions-format"></a>
</h3>
<p>The modularity of <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
is enabled by standardizing how the genomic regions and samples are
represented in the input and output of all its functions. We chose to
use a tibble because this allows you to easily plot and explore the
data. The columns are described in the table below. Only ‘chrom’,
‘start’, ‘end’, ‘sample_name’ are required to use <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>,
but you have more possibilities for <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
and <a href="#filter-regions"><code>filter_regions</code></a> if you can
provide the optional ‘score’ and ‘center’ columns. If you can’t provide
data for the optional columns, <code>prepare_input_regions</code> adds
sensible defaults for you. We recommend that you create this standard
tibble using <code>prepare_input_regions</code> (see section “Prepare
input data”), but you can also prepare it yourself (see section “<a href="#load-from-pre-loaded-tibble">Load from pre-loaded
tibble</a>”)</p>
<table class="table">
<colgroup>
<col width="17%">
<col width="18%">
<col width="21%">
<col width="42%">
</colgroup>
<thead><tr class="header">
<th align="left">Column name</th>
<th align="left">Requirement</th>
<th align="left">Content</th>
<th align="left">Details</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">‘chrom’</td>
<td align="left">required</td>
<td align="left">Name of the chromosome.</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">‘start’</td>
<td align="left">required</td>
<td align="left">Start coordinate of the genomic region.</td>
<td align="left">1-based coordinate system, NOT like BED files which are
0-based.</td>
</tr>
<tr class="odd">
<td align="left">‘end’</td>
<td align="left">required</td>
<td align="left">End coordinate of the genomic region.</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">‘name’</td>
<td align="left">optional</td>
<td align="left">Can be any unique identifier for a sample and genomic
region, but prepare_input_regions concatenates the ‘sample_name’ and row
number.</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">‘score’</td>
<td align="left">optional</td>
<td align="left">Selected enrichment value used to rank genomic
regions.</td>
<td align="left">Bigger values are more important. For example, qValue
from Macs2, -log10FDR from another method, or fold enrichment over
background computed from your favorite method. If not provided in input
data, <code>prepare_input_regions</code> is creating this column and
populating it with ‘0’.</td>
</tr>
<tr class="even">
<td align="left">‘strand’</td>
<td align="left">optional</td>
<td align="left">Value to define on which strand genomic region is
located.</td>
<td align="left">Values are ‘+’, ‘-’, or ‘.’. If not provided in input
data, <code>prepare_input_regions</code> is creating this column and
populating it with ‘.’.</td>
</tr>
<tr class="odd">
<td align="left">‘center’</td>
<td align="left">optional</td>
<td align="left">Absolute genomic coordinate of the center of the
region.</td>
<td align="left">Note that another common representation of ‘center’ is
to report the number of base pairs of the summit from the ‘start’
coordinate. We call this value ‘summit’ instead of ‘center’. You can use
<code>prepare_input_regions</code> to create ‘center’ from ‘summit’. If
not provided in input data, <code>prepare_input_regions</code> is
creating this column as the midpoint of each region.</td>
</tr>
<tr class="even">
<td align="left">‘sample_name’</td>
<td align="left">required</td>
<td align="left">Unique identifier for each sample (provided in the
input data).</td>
<td align="left">Avoid special characters like spaces or tabs.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="install-peakcombiner">Install <code>peakCombiner</code><a class="anchor" aria-label="anchor" href="#install-peakcombiner"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
can be installed from Bioconductor using the <em><a href="https://CRAN.R-project.org/package=BiocManager" class="external-link">BiocManager</a></em>
package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"peakCombiner"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="an-overview-of-a-complete-run">An overview of a complete run<a class="anchor" aria-label="anchor" href="#an-overview-of-a-complete-run"></a>
</h2>
<p>Here you find an overview of a complete run of the <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
recommended workflow.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/load_all.html" class="external-link">load_all</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Loading <span style="color: #0000BB;">peakCombiner</span></span></span>
<span><span class="co">#library("peakCombiner")</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_data_tibble"</span><span class="op">)</span></span>
<span><span class="va">syn_data_tibble</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 55 × 8</span></span></span>
<span><span class="co">#&gt;    chrom start   end name  score strand center sample_name   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1    200   900 <span style="color: #BB0000;">NA</span>      100 .         500 treatment_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1      1   900 <span style="color: #BB0000;">NA</span>       97 .         500 treatment_rep3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    101   300 <span style="color: #BB0000;">NA</span>       94 .         200 control_rep2  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    301   900 <span style="color: #BB0000;">NA</span>       94 .         500 control_rep2  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    201   900 <span style="color: #BB0000;">NA</span>      100 .         500 treatment_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    301   900 <span style="color: #BB0000;">NA</span>       98 .         600 treatment_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    301  <span style="text-decoration: underline;">1</span>000 <span style="color: #BB0000;">NA</span>       96 .         600 control_rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1    301  <span style="text-decoration: underline;">1</span>100 <span style="color: #BB0000;">NA</span>       93 .         500 control_rep3  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1   <span style="text-decoration: underline;">1</span>301  <span style="text-decoration: underline;">1</span>600 <span style="color: #BB0000;">NA</span>       97 .        <span style="text-decoration: underline;">1</span>400 treatment_rep3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1   <span style="text-decoration: underline;">1</span>901  <span style="text-decoration: underline;">2</span>200 <span style="color: #BB0000;">NA</span>       98 .        <span style="text-decoration: underline;">2</span>000 treatment_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 45 more rows</span></span></span>
<span></span>
<span><span class="va">data_prepared</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepare_input_regions.html">prepare_input_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">syn_data_tibble</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_center_expand</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_prepared</span>,</span>
<span>  show_message <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  exclude_by_blacklist <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">consensus_peak</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  combined_sample_name <span class="op">=</span> <span class="st">"my_new_sample_name"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">consensus_final</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">consensus_peak</span>,</span>
<span>  expand_by <span class="op">=</span> <span class="fl">350</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">consensus_final</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name                  score strand center sample_name     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     150   850 my_new_sample_name|1    100 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">2</span>450  <span style="text-decoration: underline;">3</span>150 my_new_sample_name|2     99 .        <span style="text-decoration: underline;">2</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 my_new_sample_name|3     95 .        <span style="text-decoration: underline;">3</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>450  <span style="text-decoration: underline;">5</span>150 my_new_sample_name|4     98 .        <span style="text-decoration: underline;">4</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>750  <span style="text-decoration: underline;">6</span>450 my_new_sample_name|5     30 .        <span style="text-decoration: underline;">6</span>100 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 my_new_sample_name|6     98 .        <span style="text-decoration: underline;">7</span>000 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2     150   850 my_new_sample_name|7     50 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    150   850 my_new_sample_name|8     95 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 my_new_sample_name|9     35 .         600 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   150   850 my_new_sample_name|10    30 .         500 my_new_sample_n…</span></span></code></pre></div>
<p>Please note that the message occurring during the <a href="#filter-regions"><code>filter_regions</code></a> is expected if
chromosome names between input sample and blacklist are not absolutely
identical.</p>
<p>Finally we export the resulting consensus regions tibble and save it
as BED-file. To establish a classical BED-file we remove here the column
names.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">rtracklayer</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/BEDFile-class.html" class="external-link">export.bed</a></span><span class="op">(</span><span class="va">consensus_final</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html" class="external-link">here</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"/lists/consensus_regions.bed"</span><span class="op">)</span>, format <span class="op">=</span> <span class="st">"bed"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="prepare-input-data">Prepare input data<a class="anchor" aria-label="anchor" href="#prepare-input-data"></a>
</h2>
<p>In this section, we explain how to prepare the accepted, standardized
input files and how to add needed features to these files.
<code>prepare_input_regions</code> is the mandatory first step to
prepare your input data in the format needed for all of the following
steps in <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
(for details see section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”). This function accepts three types of input formats:</p>
<ul>
<li>a tibble of genomic regions and specifically named columns,</li>
<li>a GenomicRanges object of genomic regions and specifically named
columns, or</li>
<li>a tibble (sample sheet) that specifies file paths of BED-like files
for a set of samples (e.g. standard output files from peak callers like
‘narrowPeak’ files).</li>
</ul>
<p><em>Please note</em> that the function
<code>prepare_input_regions</code> also has a filtering step, which
automatically checks for genomic regions with the same values in the
columns ‘chrom’, ‘start’, ‘end’ and ‘sample_name’ and filters for the
strongest enriched summit (based on the ‘score’ values) per region.
Additional summits are removed. Regions from peak callers can have
multiple summits annotated per enriched genomic regions. This step is
<em>not optional</em>.</p>
<p><code>prepare_input_regions</code> returns the data that you need for
all your future work as a tibble having the structure as described in
the section “<a href="#standard-genomic-regions-format">Standard genomic
regions format</a>”.</p>
<div class="section level3">
<h3 id="quick-start">Quick start<a class="anchor" aria-label="anchor" href="#quick-start"></a>
</h3>
<p>This is for you if you want to jump right in with some code. One of
the easiest ways (and the way we recommend unless you are an advanced
user) is to provide <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
with the list of samples and files paths for the peak files in BED-like
format, and some information about the format of the peak files.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_sample_sheet"</span><span class="op">)</span></span>
<span><span class="va">syn_sample_sheet</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   sample_name file_path                              file_format score_colname</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control01   data-raw/syn_control_rep1.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control02   data-raw/syn_control_rep2.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control03   data-raw/syn_control_rep3.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment01 data-raw/syn_treatment_rep1.narrowPeak narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment02 data-raw/syn_treatment_rep2.narrowPeak narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment03 data-raw/syn_treatment_rep3.narrowPeak narrowPeak  qValue</span></span></code></pre></div>
<p>This is the expected structure of a sample_sheet.</p>
<p>If you already have your BED-like region files loaded into your R
session, you can alternatively provide them to
<code>prepare_input_regions</code> as a single GRanges or tibble object
with genomic regions and named columns (see section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”) that, importantly, include a unique sample identifier
(‘sample_name’). Here, we show as an example the tibble object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_data_control01"</span><span class="op">)</span></span>
<span><span class="va">syn_data_control01</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 6</span></span></span>
<span><span class="co">#&gt;    chrom  start   end score strand center</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     301  <span style="text-decoration: underline;">1</span>000    96 .         600</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400    95 .        <span style="text-decoration: underline;">3</span>800</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300    46 .        <span style="text-decoration: underline;">4</span>900</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300    45 .        <span style="text-decoration: underline;">5</span>100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100    26 .        <span style="text-decoration: underline;">5</span>700</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400    25 .        <span style="text-decoration: underline;">7</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000    75 .         600</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000    50 .         600</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Chr2     101   800    80 .         700</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000    30 .         600</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100    20 .         600</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_data_treatment01"</span><span class="op">)</span></span>
<span><span class="va">syn_data_treatment01</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 6</span></span></span>
<span><span class="co">#&gt;    chrom  start   end score strand center</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     200   900   100 .         500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     201   900   100 .         500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200    99 .        <span style="text-decoration: underline;">2</span>800</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200    60 .        <span style="text-decoration: underline;">4</span>700</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200    59 .        <span style="text-decoration: underline;">5</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300    30 .        <span style="text-decoration: underline;">6</span>100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400    29 .        <span style="text-decoration: underline;">7</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    201   900    95 .         500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2     201   900    50 .         500</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   201   900    30 .         500</span></span></code></pre></div>
<p>Let’s combine the two tibbles.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">combined_input</span> <span class="op">&lt;-</span> <span class="va">syn_data_control01</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"control-rep1"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">syn_data_treatment01</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"treatment-rep1"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">combined_input</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt;   sample_name    num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control-rep1            11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> treatment-rep1          10</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepare_input_regions.html">prepare_input_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">combined_input</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 19 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name              score strand center sample_name   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control-rep1|9       80 .         700 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control-rep1|1       96 .         600 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control-rep1|2       95 .        <span style="text-decoration: underline;">3</span>800 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control-rep1|3       46 .        <span style="text-decoration: underline;">4</span>900 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control-rep1|5       26 .        <span style="text-decoration: underline;">5</span>700 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control-rep1|6       25 .        <span style="text-decoration: underline;">7</span>000 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control-rep1|7       75 .         600 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control-rep1|8       50 .         600 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control-rep1|10      30 .         600 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control-rep1|11      20 .         600 control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr1     200   900 treatment-rep1|1    100 .         500 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr1     201   900 treatment-rep1|2    100 .         500 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200 treatment-rep1|3     99 .        <span style="text-decoration: underline;">2</span>800 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 treatment-rep1|4     60 .        <span style="text-decoration: underline;">4</span>700 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300 treatment-rep1|6     30 .        <span style="text-decoration: underline;">6</span>100 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 treatment-rep1|7     29 .        <span style="text-decoration: underline;">7</span>000 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> chr10    201   900 treatment-rep1|8     95 .         500 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> chr2     201   900 treatment-rep1|9     50 .         500 treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> chr4-2   201   900 treatment-rep1|10    30 .         500 treatment-rep1</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-from-sample-sheet">Load from sample sheet<a class="anchor" aria-label="anchor" href="#load-from-sample-sheet"></a>
</h3>
<p>We recommend that you use a sample sheet to prepare your data,
especially if you have ‘narrowPeak’ or ‘broadPeak’ files.
<code>prepare_input_regions</code> loads in the genomic regions file and
formats them so that they are ready to be used in peakCombiner’s other
functions. The sample sheet has three required columns and one optional
column, and is described as follows.</p>
<table class="table">
<colgroup>
<col width="18%">
<col width="18%">
<col width="31%">
<col width="31%">
</colgroup>
<thead><tr class="header">
<th align="left">Column name</th>
<th align="left">Requirement</th>
<th align="left">Column content</th>
<th align="left">Details</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">‘sample_name’</td>
<td align="left">required</td>
<td align="left">Unique name for each input sample.</td>
<td align="left">The user defines this value. Please avoid special
characters (‘space’, ‘-’, etc.).</td>
</tr>
<tr class="even">
<td align="left">‘path_peaks’</td>
<td align="left">required</td>
<td align="left">Path to the file in which the genomic regions are
stored.</td>
<td align="left">For example, the path to a ‘BED’ or ‘narrowPeak’ file.
Can be a relative path.</td>
</tr>
<tr class="odd">
<td align="left">‘file_format’</td>
<td align="left">required</td>
<td align="left">Recognized values are: ‘BED’, ‘narrowPeak’, and
‘broadPeak’.</td>
<td align="left">Used to correctly name columns from input files. Must
be the same for all samples loaded at once.</td>
</tr>
<tr class="even">
<td align="left">‘score_column’</td>
<td align="left">optional</td>
<td align="left">The exact name of the original column to be used as
score or the number of the column having the the metric used to rank
peak importance, where bigger values are more important.</td>
<td align="left">If not provided, column ‘9’ will be used for
‘narrowPeak’ or ‘broadPeak’ file formats. Column ‘9’ corresponds to the
<code>qValue</code> as described in the UCSC documentation <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12" class="external-link">here</a>.</td>
</tr>
</tbody>
</table>
<p>The following code illustrates how you prepare an accepted sample
sheet in R.</p>
<ul>
<li>
<p><strong>Identify paths to file.</strong></p>
<p>First, let’s get the paths to the peak files we want to use and save
it.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#file_names &lt;- list.files(</span></span>
<span><span class="co">#  paste0(here::here(), "/inst/syndata"),</span></span>
<span><span class="co">#  pattern = ".narrowPeak$",</span></span>
<span><span class="co">#  full.names = TRUE</span></span>
<span><span class="co">#)</span></span>
<span><span class="va">file_names</span> </span></code></pre></div>
</li>
<li>
<p><strong>Create a sample sheet.</strong></p>
<p>Next, we create a tibble (named ‘sample_sheet’) with the correct
column names (‘sample_name’, ‘file_path’, ‘file_format’,
‘score_colname’) to load in our data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_sheet</span> <span class="op">&lt;-</span> <span class="fu">peakCombiner</span><span class="fu">::</span><span class="va"><a href="../reference/syn_sample_sheet.html">syn_sample_sheet</a></span></span>
<span></span>
<span><span class="va">sample_sheet</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   sample_name file_path                              file_format score_colname</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control01   data-raw/syn_control_rep1.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control02   data-raw/syn_control_rep2.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control03   data-raw/syn_control_rep3.narrowPeak   narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment01 data-raw/syn_treatment_rep1.narrowPeak narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment02 data-raw/syn_treatment_rep2.narrowPeak narrowPeak  qValue       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment03 data-raw/syn_treatment_rep3.narrowPeak narrowPeak  qValue</span></span></code></pre></div>
<p>With this step you create a new tibble containing all the required
information to run <code>prepare_input_regions</code>.</p>
</li>
<li>
<p><strong>Prepare data from the sample sheet.</strong></p>
<p>Now we use the prepared tibble (sample_sheet) and add it as argument
<code>data</code> into the function
<code>prepare_input_regions</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#prepare_input_regions(</span></span>
<span><span class="co">#  data = sample_sheet,</span></span>
<span><span class="co">#  show_messages = FALSE</span></span>
<span><span class="co">#)</span></span></code></pre></div>
</li>
</ul>
<p>This returned value is a tibble that contains all required
information formatted correctly in order to use the downstream functions
within <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
For more information about its structure, go back to the “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>” section.</p>
</div>
<div class="section level3">
<h3 id="load-from-pre-loaded-tibble">Load from pre-loaded tibble<a class="anchor" aria-label="anchor" href="#load-from-pre-loaded-tibble"></a>
</h3>
<p>If you have already been working with your genomic regions from
within an R session, you may have them pre-loaded as a tibble (this
section) or as a GRanges object (next section: <a href="#load-from-genomicranges-object">Load from GenomicRanges
object</a>). This could be either that per sample one tibble exists or
that you ran <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
before on two data sets and now want to combine these. In this section
we show how you prepare your input data from pre-loaded tibbles.</p>
<p>We start by loading in two of our synthetic data sets in ‘narrowPeak’
format, a common format generated by peak callers like MACS. Most
BED-like files (including ‘narrowPeak’ files) don’t include column names
in the file so you will have to name them yourself using the standard
naming conventions as described by UCSC for <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1" class="external-link">BED</a> or <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12" class="external-link">narrowPeak</a>
or <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format13" class="external-link">broadPeak</a>.
As we know what to expect in each column we can name the columns
correctly: ‘chrom’ (X1), ‘start’ (X2), ‘end’ (X3), ‘name’ (X4), ‘score’
(X5) and the ‘strand’ (X6). To be clear, the names that you define for
the columns are used by <code>prepare_input_regions</code> to property
format the data.</p>
<p>Columns named ‘chrom’, ‘start’, ‘end’, ‘name’, ‘score’, ‘strand’,
‘center’ and ‘sample_name’ are maintained. <em>Please note</em> that
within <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
we call columns with relative information about the peak summit ‘summit’
and with absolute values ‘center’. So, if a column named ‘summit’ is
provided, it is used to populate a column named ‘center’. All other
columns are dropped at the end of
<code>prepare_input_regions</code>.</p>
<ul>
<li>
<p><strong>Prepare input files</strong></p>
<p>Now let’s load the first narrowPeak file. Note that the columns are
named already correctly and we expect this from your data as well.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">syn_control_rep1_narrowPeak</span><span class="op">)</span></span>
<span><span class="va">syn_control_rep1_narrowPeak</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 10</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name  score strand signalValue pValue qValue  peak</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     301  <span style="text-decoration: underline;">1</span>000 .         0 .               96     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 .         0 .               95     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               46     -<span style="color: #BB0000;">1</span>   3.33   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               45     -<span style="color: #BB0000;">1</span>   3.31   399</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 .         0 .               26     -<span style="color: #BB0000;">1</span>   2.83    99</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 .         0 .               25     -<span style="color: #BB0000;">1</span>   2.80   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 .         0 .               75     -<span style="color: #BB0000;">1</span>   3.75   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 .         0 .               50     -<span style="color: #BB0000;">1</span>   3.40   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Chr2     101   800 .         0 .               80     -<span style="color: #BB0000;">1</span>   3.81   599</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 .         0 .               20     -<span style="color: #BB0000;">1</span>   2.60   199</span></span></code></pre></div>
<p>And the second file.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">syn_treatment_rep1_narrowPeak</span><span class="op">)</span></span>
<span><span class="va">syn_treatment_rep1_narrowPeak</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 10</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name  score strand signalValue pValue qValue  peak</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     200   900 .         0 .              100     -<span style="color: #BB0000;">1</span>   4      300</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     201   900 .         0 .              100     -<span style="color: #BB0000;">1</span>   4      299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200 .         0 .               99     -<span style="color: #BB0000;">1</span>   3.99   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 .         0 .               60     -<span style="color: #BB0000;">1</span>   3.56   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 .         0 .               59     -<span style="color: #BB0000;">1</span>   3.54   499</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 .         0 .               29     -<span style="color: #BB0000;">1</span>   2.92   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    201   900 .         0 .               95     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2     201   900 .         0 .               50     -<span style="color: #BB0000;">1</span>   3.40   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   201   900 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span></code></pre></div>
</li>
<li>
<p><strong>Add column ‘sample_name’</strong></p>
<p>Now we add a column named ‘sample_name’ to each of our tibbles.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">control</span> <span class="op">&lt;-</span> <span class="va">syn_control_rep1_narrowPeak</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"control-rep1"</span><span class="op">)</span></span>
<span><span class="va">control</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 11 × 11</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name  score strand signalValue pValue qValue  peak</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     301  <span style="text-decoration: underline;">1</span>000 .         0 .               96     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 .         0 .               95     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               46     -<span style="color: #BB0000;">1</span>   3.33   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               45     -<span style="color: #BB0000;">1</span>   3.31   399</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 .         0 .               26     -<span style="color: #BB0000;">1</span>   2.83    99</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 .         0 .               25     -<span style="color: #BB0000;">1</span>   2.80   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 .         0 .               75     -<span style="color: #BB0000;">1</span>   3.75   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 .         0 .               50     -<span style="color: #BB0000;">1</span>   3.40   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Chr2     101   800 .         0 .               80     -<span style="color: #BB0000;">1</span>   3.81   599</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 .         0 .               20     -<span style="color: #BB0000;">1</span>   2.60   199</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: sample_name &lt;chr&gt;</span></span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treatment</span> <span class="op">&lt;-</span> <span class="va">syn_treatment_rep1_narrowPeak</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_name <span class="op">=</span> <span class="st">"treatment-rep1"</span><span class="op">)</span></span>
<span><span class="va">treatment</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 11</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name  score strand signalValue pValue qValue  peak</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     200   900 .         0 .              100     -<span style="color: #BB0000;">1</span>   4      300</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     201   900 .         0 .              100     -<span style="color: #BB0000;">1</span>   4      299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200 .         0 .               99     -<span style="color: #BB0000;">1</span>   3.99   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 .         0 .               60     -<span style="color: #BB0000;">1</span>   3.56   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 .         0 .               59     -<span style="color: #BB0000;">1</span>   3.54   499</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 .         0 .               29     -<span style="color: #BB0000;">1</span>   2.92   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    201   900 .         0 .               95     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2     201   900 .         0 .               50     -<span style="color: #BB0000;">1</span>   3.40   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   201   900 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: sample_name &lt;chr&gt;</span></span></span></code></pre></div>
</li>
<li>
<p><strong>Combine multiple tibbles</strong></p>
<p>Finally, combine the multiple input tibbles into one.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">combined_input</span> <span class="op">&lt;-</span> <span class="va">control</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">treatment</span><span class="op">)</span></span>
<span><span class="va">combined_input</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 21 × 11</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name  score strand signalValue pValue qValue  peak</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     301  <span style="text-decoration: underline;">1</span>000 .         0 .               96     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 .         0 .               95     -<span style="color: #BB0000;">1</span>   3.96   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               46     -<span style="color: #BB0000;">1</span>   3.33   199</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 .         0 .               45     -<span style="color: #BB0000;">1</span>   3.31   399</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 .         0 .               26     -<span style="color: #BB0000;">1</span>   2.83    99</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 .         0 .               25     -<span style="color: #BB0000;">1</span>   2.80   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 .         0 .               75     -<span style="color: #BB0000;">1</span>   3.75   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 .         0 .               50     -<span style="color: #BB0000;">1</span>   3.40   299</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Chr2     101   800 .         0 .               80     -<span style="color: #BB0000;">1</span>   3.81   599</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 .         0 .               30     -<span style="color: #BB0000;">1</span>   2.95   299</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 11 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: sample_name &lt;chr&gt;</span></span></span></code></pre></div>
<pre><code>And check how many rows we have now for each sample.</code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">combined_input</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"number_of_entries"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   sample_name [2]</span></span></span>
<span><span class="co">#&gt;   sample_name    number_of_entries</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control-rep1                  11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> treatment-rep1                10</span></span></code></pre></div>
<p>Both ’sample_name’s are found, so we know that we have successfully
combined the data sets.</p>
</li>
<li>
<p><strong>Prepare data from the pre-loaded tibble</strong></p>
<p>After preparing the pre-loaded tibble, we run the function
<code>prepare_input_regions</code> and use the tibble in the parameter
<code>data</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepare_input_regions.html">prepare_input_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">combined_input</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 19 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name              score strand center sample_name   </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control-rep1|9        0 .        450. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control-rep1|1        0 .        650. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control-rep1|2        0 .       <span style="text-decoration: underline;">3</span>950. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control-rep1|3        0 .       <span style="text-decoration: underline;">5</span>000. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control-rep1|5        0 .       <span style="text-decoration: underline;">5</span>850. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control-rep1|6        0 .       <span style="text-decoration: underline;">7</span>050. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control-rep1|7        0 .        650. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control-rep1|8        0 .        650. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control-rep1|10       0 .        650. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control-rep1|11       0 .        750. control-rep1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr1     200   900 treatment-rep1|1      0 .        550  treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr1     201   900 treatment-rep1|2      0 .        550. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr1    <span style="text-decoration: underline;">2</span>601  <span style="text-decoration: underline;">3</span>200 treatment-rep1|3      0 .       <span style="text-decoration: underline;">2</span>900. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr1    <span style="text-decoration: underline;">4</span>501  <span style="text-decoration: underline;">5</span>200 treatment-rep1|5      0 .       <span style="text-decoration: underline;">4</span>850. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> chr1    <span style="text-decoration: underline;">5</span>801  <span style="text-decoration: underline;">6</span>300 treatment-rep1|6      0 .       <span style="text-decoration: underline;">6</span>050. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 treatment-rep1|7      0 .       <span style="text-decoration: underline;">7</span>050. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> chr10    201   900 treatment-rep1|8      0 .        550. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> chr2     201   900 treatment-rep1|9      0 .        550. treatment-rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> chr4-2   201   900 treatment-rep1|10     0 .        550. treatment-rep1</span></span></code></pre></div>
</li>
</ul>
<p>The output tibble from <code>prepare_input_data</code> can now be
used for your next steps with <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
For details about the accepted file structure see section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”.</p>
</div>
<div class="section level3">
<h3 id="load-from-genomicranges-object">Load from GenomicRanges object<a class="anchor" aria-label="anchor" href="#load-from-genomicranges-object"></a>
</h3>
<p>In memory GenomicRanges object listing the genomic regions in a
sample. This object is very similar to the tibble above, except that
<code>chrom</code>, <code>start</code>, and <code>end</code> are instead
described using the <code>GenomicRanges</code> nomenclature (See <a href="https://bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html" class="external-link">here</a>
for details ).</p>
<ul>
<li>
<p><strong>Load in a GenomicRanges object</strong></p>
<p>As first step we load the provided synthetic data originating from a
GenomicRanges object.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_data_granges"</span><span class="op">)</span></span>
<span><span class="va">syn_data_granges</span></span>
<span><span class="co">#&gt;    seqnames start  end width strand score center    sample_name</span></span>
<span><span class="co">#&gt; 1      chr1   200  900   701      *   100    500 treatment_rep1</span></span>
<span><span class="co">#&gt; 2      chr1     1  900   900      *    97    500 treatment_rep3</span></span>
<span><span class="co">#&gt; 3      chr1   101  300   200      *    94    200   control_rep2</span></span>
<span><span class="co">#&gt; 4      chr1   301  900   600      *    94    500   control_rep2</span></span>
<span><span class="co">#&gt; 5      chr1   201  900   700      *   100    500 treatment_rep1</span></span>
<span><span class="co">#&gt; 6      chr1   301  900   600      *    98    600 treatment_rep2</span></span>
<span><span class="co">#&gt; 7      chr1   301 1000   700      *    96    600   control_rep1</span></span>
<span><span class="co">#&gt; 8      chr1   301 1100   800      *    93    500   control_rep3</span></span>
<span><span class="co">#&gt; 9      chr1  1301 1600   300      *    97   1400 treatment_rep3</span></span>
<span><span class="co">#&gt; 10     chr1  1901 2200   300      *    98   2000 treatment_rep2</span></span>
<span><span class="co">#&gt; 11     chr1  2501 3100   600      *    97   2800 treatment_rep3</span></span>
<span><span class="co">#&gt; 12     chr1  2501 3400   900      *    98   3000 treatment_rep2</span></span>
<span><span class="co">#&gt; 13     chr1  2601 3200   600      *    99   2800 treatment_rep1</span></span>
<span><span class="co">#&gt; 14     chr1  3501 4200   700      *    44   3800   control_rep2</span></span>
<span><span class="co">#&gt; 15     chr1  3501 4400   900      *    95   3800   control_rep1</span></span>
<span><span class="co">#&gt; 16     chr1  3601 4400   800      *    43   3900   control_rep3</span></span>
<span><span class="co">#&gt; 17     chr1  4501 5000   500      *    97   4800 treatment_rep3</span></span>
<span><span class="co">#&gt; 18     chr1  4501 5200   700      *    60   4700 treatment_rep1</span></span>
<span><span class="co">#&gt; 19     chr1  4501 5200   700      *    59   5000 treatment_rep1</span></span>
<span><span class="co">#&gt; 20     chr1  4501 5300   800      *    98   4800 treatment_rep2</span></span>
<span><span class="co">#&gt; 21     chr1  4501 5300   800      *    98   5100 treatment_rep2</span></span>
<span><span class="co">#&gt; 22     chr1  4601 5100   500      *    93   4900   control_rep3</span></span>
<span><span class="co">#&gt; 23     chr1  4601 5200   600      *    94   4800   control_rep2</span></span>
<span><span class="co">#&gt; 24     chr1  4701 5300   600      *    46   4900   control_rep1</span></span>
<span><span class="co">#&gt; 25     chr1  4701 5300   600      *    45   5100   control_rep1</span></span>
<span><span class="co">#&gt; 26     chr1  5601 6100   500      *    26   5700   control_rep1</span></span>
<span><span class="co">#&gt; 27     chr1  5701 6400   700      *    98   6200 treatment_rep2</span></span>
<span><span class="co">#&gt; 28     chr1  5801 6300   500      *    30   6100 treatment_rep1</span></span>
<span><span class="co">#&gt; 29     chr1  6701 7400   700      *    25   7000   control_rep1</span></span>
<span><span class="co">#&gt; 30     chr1  6701 7400   700      *    44   7000   control_rep2</span></span>
<span><span class="co">#&gt; 31     chr1  6701 7400   700      *    43   7000   control_rep3</span></span>
<span><span class="co">#&gt; 32     chr1  6701 7400   700      *    29   7000 treatment_rep1</span></span>
<span><span class="co">#&gt; 33     chr1  6701 7400   700      *    98   7000 treatment_rep2</span></span>
<span><span class="co">#&gt; 34     chr1  6701 7400   700      *    97   7000 treatment_rep3</span></span>
<span><span class="co">#&gt; 35    chr10   101  800   700      *    95    400   control_rep2</span></span>
<span><span class="co">#&gt; 36    chr10   101  900   800      *    80    500 treatment_rep3</span></span>
<span><span class="co">#&gt; 37    chr10   201  900   700      *    95    500 treatment_rep1</span></span>
<span><span class="co">#&gt; 38    chr10   301 1000   700      *    75    600   control_rep1</span></span>
<span><span class="co">#&gt; 39    chr10   301 1000   700      *    90    600 treatment_rep2</span></span>
<span><span class="co">#&gt; 40    chr10   301 1000   700      *    90    600   control_rep3</span></span>
<span><span class="co">#&gt; 41     chr2   101  800   700      *    30    400   control_rep2</span></span>
<span><span class="co">#&gt; 42     chr2   101  900   800      *    10    500 treatment_rep3</span></span>
<span><span class="co">#&gt; 43     chr2   201  900   700      *    50    500 treatment_rep1</span></span>
<span><span class="co">#&gt; 44     chr2   301 1000   700      *    50    600   control_rep1</span></span>
<span><span class="co">#&gt; 45     chr2   301 1000   700      *    10    600   control_rep3</span></span>
<span><span class="co">#&gt; 46     chr2   301 1000   700      *    30    600 treatment_rep2</span></span>
<span><span class="co">#&gt; 47     Chr2   101  800   700      *    80    700   control_rep1</span></span>
<span><span class="co">#&gt; 48   chr4 2   301 1000   700      *    30    600   control_rep1</span></span>
<span><span class="co">#&gt; 49   chr4 2   101  800   700      *    25    400   control_rep2</span></span>
<span><span class="co">#&gt; 50   chr4 2   301 1000   700      *    35    600   control_rep3</span></span>
<span><span class="co">#&gt; 51   chr4-2   401 1100   700      *    20    600   control_rep1</span></span>
<span><span class="co">#&gt; 52   chr4-2   201  900   700      *    30    500 treatment_rep1</span></span>
<span><span class="co">#&gt; 53   chr4?2   101  900   800      *    25    400 treatment_rep3</span></span>
<span><span class="co">#&gt; 54   chr4|2   101  800   700      *    80    400   control_rep2</span></span>
<span><span class="co">#&gt; 55    chr42   301 1000   700      *    90    600 treatment_rep2</span></span></code></pre></div>
<p>The column names are based on its original GRanges file format. This
allows us to easily transform it into a GRanges object. Note that
normally we expect you to have the GRanges object pre-loaded and want to
use the <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
on this data set. For the purpose of showing you how a accepted GRanges
object has to be structured we transform it here.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GenomicRanges_data</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span></span>
<span>  <span class="va">syn_data_granges</span>,</span>
<span>  keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">GenomicRanges_data</span></span>
<span><span class="co">#&gt; GRanges object with 55 ranges and 3 metadata columns:</span></span>
<span><span class="co">#&gt;        seqnames    ranges strand |     score    center    sample_name</span></span>
<span><span class="co">#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;    &lt;character&gt;</span></span>
<span><span class="co">#&gt;    [1]     chr1   200-900      * |       100       500 treatment_rep1</span></span>
<span><span class="co">#&gt;    [2]     chr1     1-900      * |        97       500 treatment_rep3</span></span>
<span><span class="co">#&gt;    [3]     chr1   101-300      * |        94       200   control_rep2</span></span>
<span><span class="co">#&gt;    [4]     chr1   301-900      * |        94       500   control_rep2</span></span>
<span><span class="co">#&gt;    [5]     chr1   201-900      * |       100       500 treatment_rep1</span></span>
<span><span class="co">#&gt;    ...      ...       ...    ... .       ...       ...            ...</span></span>
<span><span class="co">#&gt;   [51]   chr4-2  401-1100      * |        20       600   control_rep1</span></span>
<span><span class="co">#&gt;   [52]   chr4-2   201-900      * |        30       500 treatment_rep1</span></span>
<span><span class="co">#&gt;   [53]   chr4?2   101-900      * |        25       400 treatment_rep3</span></span>
<span><span class="co">#&gt;   [54]   chr4|2   101-800      * |        80       400   control_rep2</span></span>
<span><span class="co">#&gt;   [55]    chr42  301-1000      * |        90       600 treatment_rep2</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 9 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</li>
<li>
<p><strong>Prepare input from GenomicRanges object</strong></p>
<p>You can simply use your GRanges object in the parameter ‘data’ and
load it in. The output tibble from <code>prepare_input_data</code> can
now be used for your next steps with <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
For details about the accepted file structure see section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepare_input_regions.html">prepare_input_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">GenomicRanges_data</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; → Column <span style="color: #00BB00;">'chrom'</span> is not class <span style="color: #0000BB;">&lt;character&gt;</span>. It will be</span></span>
<span><span class="co">#&gt; converted to class <span style="color: #0000BB;">&lt;character&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Checking whether duplicated regions exist and need to be collapsed.</span></span>
<span><span class="co">#&gt; → Checked whether duplicated regions exist and need to be collapsed.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Duplicated regions identified and collapsed to unique <span style="color: #00BB00;">chrom</span>, <span style="color: #00BB00;">start</span>, and <span style="color: #00BB00;">end</span></span></span>
<span><span class="co">#&gt;   for each sample by strongest <span style="color: #00BB00;">score</span> value.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Preparation of data finished successfully.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="explained-in-detail">Explained in detail<a class="anchor" aria-label="anchor" href="#explained-in-detail"></a>
</h3>
<p>We recommend to use the function <code>prepare_input_regions</code>
to prepare the input data in the format needed for all of the following
steps in <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
In theory you can also manually provide an expected input data when
preparing your data following the descriptions in the section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”.</p>
<div class="section level4">
<h4 id="load-from-bed-file">Load from BED file<a class="anchor" aria-label="anchor" href="#load-from-bed-file"></a>
</h4>
<p>Unlike ‘narrowPeak’ files, BED files typically do not include columns
for summits (‘summit’) or significance (‘score’) for your peaks. For
that reason, we recommend using ‘narrowPeak’ files if possible.
Sometimes you may only have ‘chrom’, ‘start’, and ‘end’ and you may
still use <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
Here we show you how to load it.</p>
<p>Lets load a BED file.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"syn_data_bed"</span><span class="op">)</span></span>
<span><span class="va">syn_data_bed</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 55 × 4</span></span></span>
<span><span class="co">#&gt;    chrom  start   end sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Chr2     101   800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 45 more rows</span></span></span></code></pre></div>
<p>When we pull the ‘sample_name’ column we see the different number of
entries for each sample name.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">syn_data_bed</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   sample_name    num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control_rep1            11</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control_rep2             9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control_rep3             7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment_rep1          10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment_rep2          10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment_rep3           8</span></span></code></pre></div>
<p>And now we use it as input for
<code>prepare_input_regions</code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/prepare_input_regions.html">prepare_input_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">syn_data_bed</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Provide input `data` is a pre-loaded <span style="color: #0000BB;">&lt;data.frame&gt;</span> with the required column</span></span>
<span><span class="co">#&gt;   names.</span></span>
<span><span class="co">#&gt; → Start preparing data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Required columns will be added if missing.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Column <span style="color: #00BB00;">'score'</span> does not exist in `data_prepared`.</span></span>
<span><span class="co">#&gt; → Column <span style="color: #00BB00;">'score'</span> is added and filled with <span style="color: #0000BB;">"0"</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Column <span style="color: #00BB00;">'strand'</span> does not exist in `data_prepared`.</span></span>
<span><span class="co">#&gt; → Column <span style="color: #00BB00;">'strand'</span> is added and filled with <span style="color: #0000BB;">"."</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Column <span style="color: #00BB00;">'summit'</span> does not exist in `data_prepared`.</span></span>
<span><span class="co">#&gt; → As no input column <span style="color: #00BB00;">'summit'</span> is found, the output column <span style="color: #00BB00;">'center'</span> has to be</span></span>
<span><span class="co">#&gt;   filled with arithmetic center of peak.</span></span>
<span><span class="co">#&gt; → Checking <span style="color: #0000BB;">&lt;class&gt;</span> and <span style="color: #0000BB;">"values"</span> of all columns.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Checking whether duplicated regions exist and need to be collapsed.</span></span>
<span><span class="co">#&gt; → Checked whether duplicated regions exist and need to be collapsed.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Duplicated regions identified and collapsed to unique <span style="color: #00BB00;">chrom</span>, <span style="color: #00BB00;">start</span>, and <span style="color: #00BB00;">end</span></span></span>
<span><span class="co">#&gt;   for each sample by strongest <span style="color: #00BB00;">score</span> value.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Preparation of data finished successfully.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     101   800 control_rep1|9      0 .        450. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     301  <span style="text-decoration: underline;">1</span>000 control_rep1|1      0 .        650. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>501  <span style="text-decoration: underline;">4</span>400 control_rep1|2      0 .       <span style="text-decoration: underline;">3</span>950. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>701  <span style="text-decoration: underline;">5</span>300 control_rep1|3      0 .       <span style="text-decoration: underline;">5</span>000. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>601  <span style="text-decoration: underline;">6</span>100 control_rep1|5      0 .       <span style="text-decoration: underline;">5</span>850. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>701  <span style="text-decoration: underline;">7</span>400 control_rep1|6      0 .       <span style="text-decoration: underline;">7</span>050. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    301  <span style="text-decoration: underline;">1</span>000 control_rep1|7      0 .        650. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     301  <span style="text-decoration: underline;">1</span>000 control_rep1|8      0 .        650. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   301  <span style="text-decoration: underline;">1</span>000 control_rep1|10     0 .        650. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   401  <span style="text-decoration: underline;">1</span>100 control_rep1|11     0 .        750. control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
<p>Please note here that the information messages are informing you
about all missing values and with which default values these columns are
populated. The ‘score’ is set to 0 as no information can be obtained
from a classical BED file about enrichment values. The column ‘strand’
is populated with the value ‘.’, representing that no strand information
is known. The ‘center’ is calculated based on the arithmetical midpoint
of each region as no ‘summit’ input column was found. The resulting
tibble can be used with all functions (<a href="#center-and-expand-regions"><code>center_expand_regions</code></a>,
<a href="#filter-regions"><code>filter_regions</code></a>, <a href="#combine-regions"><code>combine_regions</code></a>) of the package
but certain option are limited due to the missing information in the
input.</p>
<p>For instance, <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
is limited to use the ‘midpoint’ as center as no summit information is
provided (See section <a href="#center-and-expand-regions">Center and
expand regions</a>). <a href="#filter-regions"><code>filter_regions</code></a> the options
‘include_above_score_cutoff’ and ‘include_top_n_scoring’ do rely on
values in the column ‘score’ to be populated with different values (See
section <a href="#filter-regions">Filter regions</a>) and can not be
used.</p>
</div>
<div class="section level4">
<h4 id="working-in-1-based-space">Working in 1-based space<a class="anchor" aria-label="anchor" href="#working-in-1-based-space"></a>
</h4>
<p>The package <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
is expecting to work with 1-based values instead of 0-based or a mix
setup as in <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1" class="external-link">BED file</a>.
For example in <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12" class="external-link">narrowPeak</a>
files the start coordinate is included in the region, while the end
coordinate is not. This follows the classical definition of a <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1" class="external-link">BED file</a>
in UCSC, where the start is included but the end not, meaning the start
is 0-based and the end is 1-based. As normally in displaying genomic
regions browsers both coordinates are used in the 1-based way, we
decided to use for <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
exclusivity the 1-based approach. A good cheat sheet is linked <a href="https://www.biostars.org/p/84686/" class="external-link">here</a>.</p>
<p>If you load your data using a sample sheet “<a href="#load-from-sample-sheet">Load from sample sheet</a>”, <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
takes care of the details for you.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="center-and-expand-regions">Center and expand regions<a class="anchor" aria-label="anchor" href="#center-and-expand-regions"></a>
</h2>
<p>After you prepare your input data, you may want to resize your peaks
to a single, consistent size for your downstream analyses. In this case,
we recommend that you first center and expand your genomic regions by
using <code>center_expand_regions</code>. <em>Please note</em> that this
function is optional, so not required but recommended to run
<code>center_expand_regions</code> before you combine your regions.</p>
<div class="section level3">
<h3 id="quick-start-1">Quick start<a class="anchor" aria-label="anchor" href="#quick-start-1"></a>
</h3>
<p>The quickest way to get started is to call
<code>center_expand_regions</code> using just the default parameters. In
this case, <code>center_expand_regions</code> updates the ‘start’ and
‘end’ coordinates of each genomic region such that it is centered around
‘center’ and resized to the median of all input regions.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_prepared</span>,</span>
<span>  show_message <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     350  <span style="text-decoration: underline;">1</span>050 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     250   950 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>550  <span style="text-decoration: underline;">5</span>250 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>050 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    250   950 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     250   950 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   950 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
<p>The tibble you obtained has altered coordinates for the genomic
region. The center of the region is defined as the value found in the
column ‘center’ and the expansion is based on the median region size of
all input regions.</p>
</div>
<div class="section level3">
<h3 id="run-center-and-expand">Run center and expand<a class="anchor" aria-label="anchor" href="#run-center-and-expand"></a>
</h3>
<p>The expected input is the standard tibble as described previously
(See section “<a href="#standard-genomic-regions-format">Standard
genomic regions format</a>”). We suggest you prepare it with the
function <code>prepare_input_regions</code>.</p>
<ul>
<li>
<p><strong>Define the new center</strong></p>
<p>By default, <code>center_expand_regions</code> uses the existing
value in the ‘center’ column. (center_by = “center_column”) which is the
recommended approach.</p>
<p>However, you can also overwrite the existing ‘center’ value by
computing the midpoint of the genomic region (center_by = “midpoint”).
This makes sense to do only if you have loaded your data from a peak
caller that contain summits or the location of the strongest enrichment
(e.g., MACS narrowPeak), but you would rather use the midpoint.</p>
</li>
<li>
<p><strong>Define the value to expand</strong></p>
<p>By default, <code>center_expand_regions</code> (expand_by = NULL)
calculates a reasonable value for you from your data by using half of
the median regions length of all your inputdata to expand. You can
alternatively provide a numeric value. For example, when ‘expand_by’ is
set to 250 (expand_by = 250), this results in regions of size 500
centered around ‘center’ (for more details see section “<a href="#expand-from-center">Expand from ‘center’</a>”).</p>
</li>
<li>
<p><strong>Center &amp; expanding the regions</strong></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_prepared</span>,</span>
<span>  center_by <span class="op">=</span> <span class="st">"center_column"</span>,</span>
<span>  expand_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Input value for `expand_by` is <span style="color: #0000BB;">"NULL"</span>. Median of all input genomic regions is</span></span>
<span><span class="co">#&gt;   calculated and returned for expansion.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> `expand_by` was calculated from the input data and set to "<span style="color: #0000BB;">350</span>".</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Genomic regions will be expanded by 350bp in both direction.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Checking <span style="color: #0000BB;">&lt;class&gt;</span> and <span style="color: #0000BB;">"values"</span> of all columns.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Genomic regions will be centered and expanded.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Starting with expanding genomic regions from the column <span style="color: #00BB00;">center</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Genomic regions were successfully centered and expanded.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Some newly-defined genomic regions have a <span style="color: #00BB00;">start</span> coordinate below <span style="color: #0000BB;">"1"</span>.</span></span>
<span><span class="co">#&gt; → Values of <span style="color: #00BB00;">name</span> for these site: <span style="color: #0000BB;">"control_rep2|1"</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> These genomic regions were truncated to get <span style="color: #00BB00;">start</span> coordinate <span style="color: #0000BB;">"1"</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Genomic regions were successfully centered and expanded.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     350  <span style="text-decoration: underline;">1</span>050 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     250   950 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>550  <span style="text-decoration: underline;">5</span>250 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>050 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    250   950 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     250   950 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   950 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
<p>You can appreciate that values for ‘start’ and ‘end’ are changed,
while the number of input regions stays the same.</p>
</li>
<li>
<p><strong>Define the expansion value.</strong></p>
<p>Finally some examples how you can define the expansion, either
symmetrically or asymmetrically (for more details see section <a href="#expand-from-center">Expand from ‘center’</a>).</p>
</li>
</ul>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_prepared</span>,</span>
<span>  center_by <span class="op">=</span> <span class="st">"center_column"</span>,</span>
<span>  expand_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">500</span><span class="op">)</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     200  <span style="text-decoration: underline;">1</span>200 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     100  <span style="text-decoration: underline;">1</span>100 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>300  <span style="text-decoration: underline;">4</span>300 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>400  <span style="text-decoration: underline;">5</span>400 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>200  <span style="text-decoration: underline;">6</span>200 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>500  <span style="text-decoration: underline;">7</span>500 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    100  <span style="text-decoration: underline;">1</span>100 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     100  <span style="text-decoration: underline;">1</span>100 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   100  <span style="text-decoration: underline;">1</span>100 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   100  <span style="text-decoration: underline;">1</span>100 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/center_expand_regions.html">center_expand_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_prepared</span>,</span>
<span>  center_by <span class="op">=</span> <span class="st">"center_column"</span>,</span>
<span>  expand_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">1000</span><span class="op">)</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     600  <span style="text-decoration: underline;">1</span>700 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     500  <span style="text-decoration: underline;">1</span>600 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>700  <span style="text-decoration: underline;">4</span>800 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>800  <span style="text-decoration: underline;">5</span>900 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>600  <span style="text-decoration: underline;">6</span>700 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>900  <span style="text-decoration: underline;">8</span>000 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    500  <span style="text-decoration: underline;">1</span>600 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     500  <span style="text-decoration: underline;">1</span>600 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   500  <span style="text-decoration: underline;">1</span>600 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   500  <span style="text-decoration: underline;">1</span>600 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
<p>In all cases, the input regions are altered. If you are unsure how to
define this parameter or have no prior expectations, don’t specify a
value for the parameter ‘expand_by’ to enable its default behavior to
calculate the expansion as median of the input regions from input
data.</p>
</div>
<div class="section level3">
<h3 id="explained-in-details">Explained in details<a class="anchor" aria-label="anchor" href="#explained-in-details"></a>
</h3>
<p>In this section we want to go into more detail to help you define the
ideal parameters for the function <code>center_expand_regions</code>.
The expected input data structure we described early in this section “<a href="#standard-genomic-regions-format">Standard genomic regions
format</a>”.</p>
<p>After the preparation of the input data, the first step, we recommend
to center and expand the genomic regions by using
<code>center_expand_regions</code>. It is useful if you want all of your
peaks to be the same size for your downstream analyses or if you want to
use the information about the maximum enrichment in each reach (often
called ‘summit’) (stored in the column ‘center’), normally obtained by
some peak callers (e.g., MACS2). The function allows you to
automatically center your regions of interest on these summits to
capture information about the most important part within a genomic
region (e.g., TF-binding site or highest peak).</p>
<p>There are two concepts that are relevant to understand the function
<code>center_expand_regions</code>:</p>
<div class="section level4">
<h4 id="define-the-center">Define the ‘center’<a class="anchor" aria-label="anchor" href="#define-the-center"></a>
</h4>
<p>In general, there are two approaches to define a ‘center’ of a
genomic region and the function <code>center_expand_regions</code>
allows you to decide which one to use.</p>
<p>The first option for you is to use pre-defined summit information
(e.g., obtained from a peak caller like MACS2) (center_by =
center_column). Such information is provided when input regions were
identified by a peak calling tool that exactly defines where in the
region the maximum enriched signal is found. The second option is to
calculate the arithmetic midpoint of the genomic region (center_by =
midpoint). (For details see the help for
<code>prepare_input_regions</code>).</p>
</div>
<div class="section level4">
<h4 id="expand-from-center">Expand from ‘center’<a class="anchor" aria-label="anchor" href="#expand-from-center"></a>
</h4>
<p>The second factor you have to think about is by how many nucleotides
you want to expand from the center to re-define your genomic regions.
The function <code>center_expand_regions</code> allows you to either use
the input data to calculate the expansion value or to provide one or two
numeric values to expand. Your decision is strongly impacted by the type
of signal you are looking for. Traditionally, ATAC-seq, transcription
factor ChIP-seq and some histone marks ChIP-seq (For details see <a href="https://www.encodeproject.org/chip-seq/histone/" class="external-link">ENCODE
recommendations</a>) show a very narrow, sharp signal, and your region
size and expansion value should reflect that. On the other hand, some
histone marks are enriched on large domains showing broad patterns.
Therefore, prior knowledge of the signal you are looking for is key to
choosing the best option for this parameter. Sometimes you may not have
a prior expectation of region size, so the default in <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
is to choose a reasonable default value from the data (expand_by = NULL)
as the median input region length / 2 for expansion.</p>
<p>You can also choose to expand the genomic region from the new center
asymmetrically, by different lengths before and after the center
position (For examples see section “[Run center and expand”) by
providing a vector with two values (the ‘center’ minus the first value
defines the start and the ‘center’ plus the second defines the end).</p>
</div>
<div class="section level4">
<h4 id="correct-if-expanding-over-chromsome-boarders">Correct if expanding over chromsome boarders<a class="anchor" aria-label="anchor" href="#correct-if-expanding-over-chromsome-boarders"></a>
</h4>
<p>It is theoretically possible that centering and expanding could
result in a ‘start’ coordinate value less than 0. This of course would
be invalid. If this happens, <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
automatically replaces the negative start value with 0 and prints
feedback on how often and where this happened so you can debug if
necessary. In theory the same can happen at the ‘end’ coordinate but we
do not know the expected chromosome length and so cannot automatically
check for that. If these edge cases are critical to your downstream
analyses, we suggest that you double-check this.</p>
</div>
<div class="section level4">
<h4 id="run-function-mutliple-times">Run function mutliple times<a class="anchor" aria-label="anchor" href="#run-function-mutliple-times"></a>
</h4>
<p>If you have chosen to run <code>center_expand_regions</code> on your
input regions, you probably also want to center and expand the consensus
regions you get from <a href="#combine-regions"><code>combine_regions</code></a> so that your
final consensus set of regions all have the same length and are
well-aligned at their centers. When you combine genomic regions during
<a href="#combine-regions"><code>combine_regions</code></a>, the genomic
regions are alternated. This means that the median region size could be
different compared to your input data region size, if you try to let <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
choose the default region length from the data. For that reason, we
suggest that you save the values of the calculated expansion (which is
printed out) for the input regions and when you run the <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
again on the consensus file to use this value for the ‘expand_by’
parameter. Doing so allows you to define the consensus region length by
the exact input region length and not just a subset.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="filter-regions">Filter regions<a class="anchor" aria-label="anchor" href="#filter-regions"></a>
</h2>
<p>The next function <code>filter_regions</code> allows you to refine
the genomic regions based on four different parameters in exact the
order provided here:</p>
<ul>
<li>‘include_by_chromosome’</li>
<li>‘exclude_by_blacklist’</li>
<li>‘include_above_score_cutoff’</li>
<li>‘include_top_n_scoring’</li>
</ul>
<p>This is an <em>optional</em> step that can help you retain the most
high-quality genomic regions or reduce the overall number of genomic
regions for each sample. You can apply <code>filter_regions</code>
multiple times on the same data set, one after another, in order to
explore the effect of each individual filtering steps.</p>
<div class="section level3">
<h3 id="quick-start-2">Quick start<a class="anchor" aria-label="anchor" href="#quick-start-2"></a>
</h3>
<p>As a quick first example, you can easily exclude blacklisted hg38
regions as follows.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  exclude_by_blacklist <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_filtered</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 52 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name            score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Chr2     350  <span style="text-decoration: underline;">1</span>050 control_rep1|9     80 .         700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1     250   950 control_rep1|1     96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep1|2     95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>550  <span style="text-decoration: underline;">5</span>250 control_rep1|3     46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>050 control_rep1|5     26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep1|6     25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr10    250   950 control_rep1|7     75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2     250   950 control_rep1|8     50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 control_rep1|10    30 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   950 control_rep1|11    20 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 42 more rows</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filter-genomic-regions">Filter genomic regions<a class="anchor" aria-label="anchor" href="#filter-genomic-regions"></a>
</h3>
<p>You can filter based on four parameters:</p>
<ul>
<li><p><code>include_by_chromosome_name</code> - Retains only
chromosomes that are in the provided vector. By not including
mitochondrial, sex, or non-classical chromosomes, genomic regions found
on these chromosomes can be removed. By default, this filter is not
applied.</p></li>
<li><p><code>exclude_by_blacklist</code> - Remove ENCODE-annotated
blacklist regions for either human (<a href="https://www.encodeproject.org/files/ENCFF356LFX/" class="external-link">hg38</a>) or
mouse (<a href="https://www.encodeproject.org/files/ENCFF547MET/" class="external-link">mm10</a>).
Alternatively, a tibble can be provided listing the genomic regions to
remove (having ‘chrom’, ‘start’, and ‘end’ column names). By default,
this filter is not applied.</p></li>
<li><p><code>include_above_score_cutoff</code> - Single numeric value
that defines the threshold above which all genomic regions will be
retained based on the values in the column ‘score’. The ‘score’ column
in the <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
input data must be non-zero for this parameter to be useful. Recall that
<code>prepare_input_regions</code> sets the ‘score’ by default if
possible (e.g., it takes the value of -log10(FDR) using a ‘narrowPeak’
file from MACS as input). Importantly, applying this filter retains a
variable number of genomic regions per sample, all having a score
greater than the <code>include_above_score_cutoff</code> parameter. By
default, this filter is not applied.</p></li>
<li><p><code>include_top_n_scoring</code> - Single numeric value that
defines how many of the top scoring genomic regions (using the column
‘score’) are retained. All other genomic regions are discarded.
Importantly, applying this filter retains
<code>include_top_n_scoring</code> regions per sample, which means that
the minimum enrichment levels may vary between samples. Note that if
multiple genomic regions have the same ‘score’ cutoff value, then all of
those genomic regions are included. In this case, the number of
resulting regions retained may be a bit higher than the input parameter.
By default, this filter is not applied.</p></li>
</ul>
<p>Let’s see how what happens when all four filtering options are used
at once.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  include_by_chromosome_name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr4"</span><span class="op">)</span>,</span>
<span>  exclude_by_blacklist <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span>  include_above_score_cutoff <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>  include_top_n_scoring <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; → Checking <span style="color: #0000BB;">&lt;class&gt;</span> and <span style="color: #0000BB;">"values"</span> of all columns.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → The argument `include_by_chromosome_name` is a class <span style="color: #0000BB;">&lt;character&gt;</span> of length 3</span></span>
<span><span class="co">#&gt;   and will be used to retain matchhing chromsome names in <span style="color: #00BB00;">chrom</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Input for `include_by_chromosome_name` contains values not found in the input</span></span>
<span><span class="co">#&gt;   data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The following chromosome name you entered is not used: <span style="color: #0000BB;">"chr4"</span>.</span></span>
<span><span class="co">#&gt; → <span style="font-style: italic;">Is </span><span style="color: #0000BB; font-style: italic;">"include_by_chromosome_name"</span><span style="font-style: italic;"> correctly defined?</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Entries in <span style="color: #00BB00;">chrom</span> with the values <span style="color: #0000BB;">"chr1"</span> and <span style="color: #0000BB;">"chr2"</span> are retained.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The following 7 entries in column <span style="color: #00BB00;">chrom</span> from the input data were not</span></span>
<span><span class="co">#&gt;   retained: <span style="color: #0000BB;">"Chr2"</span>, <span style="color: #0000BB;">"chr10"</span>, <span style="color: #0000BB;">"chr4 2"</span>, <span style="color: #0000BB;">"chr4-2"</span>, <span style="color: #0000BB;">"chr4|2"</span>, <span style="color: #0000BB;">"chr42"</span>, and</span></span>
<span><span class="co">#&gt;   <span style="color: #0000BB;">"chr4?2"</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Input data was filtered to retain regions on defined chromosome.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Blacklist for annotation <span style="color: #0000BB;">"hg38"</span> will be used for filtering.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Provided input data contains chromosome names not found in blacklist.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The following input chromosomes have no match: chr10, chr11, chr12, chr13,</span></span>
<span><span class="co">#&gt;   chr14, chr15, chr16, chr17, chr18, chr19, chr20, chr21, chr22, chr3, chr4,</span></span>
<span><span class="co">#&gt;   chr5, chr6, chr7, …, chrX, and chrY.</span></span>
<span><span class="co">#&gt; → <span style="font-style: italic;">Note to user: Please doublecheck this observation.</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Input data was filtered by blacklist.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Significance in <span style="color: #00BB00;">score</span> is filtered and all regions above <span style="color: #0000BB;">2.5</span> will be retained.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> A total of 37 of 37 input regions are retained with value in <span style="color: #00BB00;">score</span> a above</span></span>
<span><span class="co">#&gt;   2.5.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Input data was filtered to retain regions with a <span style="color: #00BB00;">score</span> above the defined</span></span>
<span><span class="co">#&gt;   threshold.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The argument `include_top_n_scoring` extracted the the top 6 regions by <span style="color: #00BB00;">score</span></span></span>
<span><span class="co">#&gt;   per sample (based on the values in <span style="color: #00BB00;">sample_name</span>).</span></span>
<span><span class="co">#&gt; → The top enriched 6 regions per sample will be retained.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The argument `include_top_n_scoring` was defined as 6.</span></span>
<span><span class="co">#&gt; → The following <span style="color: #0000BB;">"sample_names"</span> contain less regions then defined by</span></span>
<span><span class="co">#&gt;   `include_top_n_scoring`: control_rep3</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No genomic regions will be removed for such samples.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Input data was filtered and the top 6 enriched regions per sample are</span></span>
<span><span class="co">#&gt;   retained.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Filtered dataset will be returned.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 35 × 8</span></span></span>
<span><span class="co">#&gt;    chrom start   end name           score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1   <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep1|6    25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>050 control_rep1|5    26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   <span style="text-decoration: underline;">4</span>550  <span style="text-decoration: underline;">5</span>250 control_rep1|3    46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr2    250   950 control_rep1|8    50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1   <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep1|2    95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    250   950 control_rep1|1    96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2     50   750 control_rep2|7    30 .         400 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1   <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep2|3    44 .        <span style="text-decoration: underline;">3</span>800 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1   <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep2|5    44 .        <span style="text-decoration: underline;">7</span>000 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1      1   550 control_rep2|1    94 .         200 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 25 more rows</span></span></span></code></pre></div>
<p>The filtering occurred in the order of the parameters and can be
described as following:</p>
<ol style="list-style-type: decimal">
<li>The data set is filtered to only include chromosomes with the names
‘chr1’, ‘chr2’ and ‘chr4’,</li>
<li>Any remaining regions overlapping with blacklisted human regions are
removed,</li>
<li>Regions having a ‘score’ above ‘2.5’ are retained, and</li>
<li>Only 6 top-scoring sites are retained per sample.</li>
</ol>
</div>
<div class="section level3">
<h3 id="explained-in-detail-1">Explained in detail<a class="anchor" aria-label="anchor" href="#explained-in-detail-1"></a>
</h3>
<p>The function <code>filter_regions</code> can help you curate the
input genomic regions when your peak-calling pipeline uses the default
options and you still need to refine the peaks further, for example when
using an automated pipeline for annotation and peak calling. The idea
here is to to provide a quick and easy way to clean-up your input data
if needed. We suggest that you run the function
<code>filter_regions</code> after the function
<code>center_expand_regions</code> just in case
<code>center_expand_regions</code> alters the genomic coordinates enough
to overlap with a blacklisted region.</p>
<p><code>filter_regions</code> allows a step-wise optimization of
selection criteria of regions of interest and can be used multiple time
on the same data set.</p>
<div class="section level4">
<h4 id="apply-the-right-filters">Apply the right filters<a class="anchor" aria-label="anchor" href="#apply-the-right-filters"></a>
</h4>
<p>Following is more detail the filtering options in <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>,
what we recommend to do. All of these parameters are optional, and by
default no filtering is done.</p>
<div class="section level5">
<h5 id="include-chromsomes">Include chromsomes<a class="anchor" aria-label="anchor" href="#include-chromsomes"></a>
</h5>
<p>Sometimes you may need to focus your analysis on a single chromosome
(e.g., to reduce the running time when testing) or set of chromosomes
(just the canonical human chromosomes). In this case,
<code>include_chromosomes</code> can help you to define the set of
chromosomes you want to focus on in your analysis.</p>
<p>In this subsection, we provide an example to identify “canonical”
human chromosomes directly from the input data, save it as vector and
use this as input for the filtering step.</p>
<ul>
<li>
<p><strong>Extract a vector of chromosomes to include</strong></p>
<p>First we extract all chromosome names from the input data.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input_chrom</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">data_center_expand</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">chrom</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">input_chrom</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 9 × 1</span></span></span>
<span><span class="co">#&gt;   chrom </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Chr2  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr1  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr10 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr2  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> chr4 2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> chr4-2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> chr4|2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span> chr42 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">9</span> chr4?2</span></span></code></pre></div>
<p>Here we see that in this data set we have some unexpected values for
chromosome names like “chr4 2”, “chr4|2” or “chr42”. Let’s modify this
vector to only keep what we consider to be the “canonical” chromosomes.
In real world human data sets, you may find names like
“chr11_KI270721v1_random” or “chrUn_GL000195v1” that you might want to
remove for your downstream analyses</p>
<p>To do so, the next step is to filter with regular expressions to
maintain only wanted chromosome names.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">include_chrom</span> <span class="op">&lt;-</span> <span class="va">input_chrom</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"^chr[0-9]$|^chr[1-2][0-9]$|^chr[XYM]"</span>, <span class="va">chrom</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">chrom</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">include_chrom</span></span>
<span><span class="co">#&gt; [1] "chr1"  "chr10" "chr2"</span></span></code></pre></div>
</li>
</ul>
<p>Finally, we can use this vector of good names in filter_regions for
the parameter <code>include_by_chromosome_name</code></p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_filtered_chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  include_by_chromosome_name <span class="op">=</span> <span class="va">include_chrom</span>,</span>
<span>  exclude_by_blacklist <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  include_above_score_cutoff <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  include_top_n_scoring <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">data_filtered_chr</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 43 × 8</span></span></span>
<span><span class="co">#&gt;    chrom start   end name           score strand center sample_name </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1    250   950 control_rep1|1    96 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1   <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep1|2    95 .        <span style="text-decoration: underline;">3</span>800 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1   <span style="text-decoration: underline;">4</span>550  <span style="text-decoration: underline;">5</span>250 control_rep1|3    46 .        <span style="text-decoration: underline;">4</span>900 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1   <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>050 control_rep1|5    26 .        <span style="text-decoration: underline;">5</span>700 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1   <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 control_rep1|6    25 .        <span style="text-decoration: underline;">7</span>000 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr10   250   950 control_rep1|7    75 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2    250   950 control_rep1|8    50 .         600 control_rep1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1      1   550 control_rep2|1    94 .         200 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1    150   850 control_rep2|2    94 .         500 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1   <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 control_rep2|3    44 .        <span style="text-decoration: underline;">3</span>800 control_rep2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 33 more rows</span></span></span></code></pre></div>
<p>Let’s confirm that we did indeed have unwanted chromosome names in
the input data.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_center_expand</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   sample_name    num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control_rep1            10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control_rep2             9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control_rep3             7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment_rep1           9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment_rep2           9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment_rep3           8</span></span></code></pre></div>
<p>And we can also confirm that only the good chromosomes remain after
this filtering step.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_filtered_chr</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">chrom</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   chrom num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr1           31</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr10           6</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr2            6</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="remove-blacklisted-sites">Remove blacklisted sites<a class="anchor" aria-label="anchor" href="#remove-blacklisted-sites"></a>
</h5>
<p>It is often recommended to exclude blacklisted genomic regions from
your analyses, and the <code>exclude_by_blacklist</code> parameter
allows you to do that easily. Genomic regions that overlap a blacklisted
region with at least 1 base are removed. We provide blacklisted regions
from ENCODE for human (<a href="https://www.encodeproject.org/files/ENCFF356LFX/" class="external-link">hg38</a>) and
mouse (<a href="https://www.encodeproject.org/files/ENCFF547MET/" class="external-link">mm10</a>). The
blacklists were manually curated by ENCODE by combing several published
blacklisted regions.</p>
<p>While we recommend to use the provided ENCODE blacklists, you can
alternatively provide your own blacklist as a tibble containing genomic
regions with columns named ‘chrom’, ‘start’ and ‘end’. In more general
terms, you can use any file with genomic locations here to remove
overlapping regions, for instance if you want to remove regions
overlapping with promoters you can use this functions here to remove
these.</p>
<p>If you would like to remove blacklisted regions from multiple
sources, you can run <code>filter_regions</code> repeatedly, for example
first to remove the ENCODE blacklisted regions and then to remove your
own blacklisted sites. We show how to do this below.</p>
</div>
<div class="section level5">
<h5 id="filter-for-enrichment-value">Filter for enrichment value<a class="anchor" aria-label="anchor" href="#filter-for-enrichment-value"></a>
</h5>
<p>Setting this parameter allows you to retain any peak that has a
bigger score than the <code>include_above_score_cutoff</code> value
based on the values in the column ‘score’. Recall that <a href="#prepare-input-data"><code>prepare_input_regions</code></a>
defines the ‘score’ by default if possible (e.g., it takes the value of
-log10(FDR) using a ‘narrowPeak’ file from MACS2 as input). As scores
might differ between experiments, we recommend that you look at the
distribution of the values in ‘score’ to help you choose the best
threshold value for <code>include_above_score_cutoff</code> for your
data set.</p>
<p>Here is some code to do that on our small, synthetic dataset. The
distribution you see will look different.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_center_expand</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="fl">0</span>,<span class="cn">NA</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_bar()`).</span></span></code></pre></div>
<p><img src="peakCombiner_files/figure-html/unnamed-chunk-37-1.png" width="80%" style="display: block; margin: auto;"></p>
<p>See that we have very few sites with low values. We use a cutoff of
35 to remove lowly enriched sites and apply this value to the filtering
step.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_filtered_cutoff</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  include_above_score_cutoff <span class="op">=</span> <span class="fl">35</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s compare the sites remaining in the input data set and after
filtering.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">data_center_expand</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 52  8</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">data_filtered_cutoff</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 39  8</span></span></code></pre></div>
<p>When we check the range of the values in ‘score’ columns we see the
effects of the filtering.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">data_center_expand</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  10 100</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">data_filtered_cutoff</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">score</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  35 100</span></span></code></pre></div>
<p>Again, we see that sites with ‘score’ 35 and below are removed.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_filtered_cutoff</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">score</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html" class="external-link">xlim</a></span><span class="op">(</span><span class="fl">0</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 1 row containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_bar()`).</span></span></code></pre></div>
<p><img src="peakCombiner_files/figure-html/unnamed-chunk-41-1.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="select-a-defined-number-of-regions">Select a defined number of regions<a class="anchor" aria-label="anchor" href="#select-a-defined-number-of-regions"></a>
</h5>
<p>You can also select a fixed number of highest scoring regions per
sample to extract the top enriched sites. An information message is
shown if any sample does not have the required number of regions left in
your input data. If your ‘score’ values vary widely between samples you
may select widely different numbers of regions using the
<code>include_above_cutoff</code>. In this case, using this approach
will help you select similar numbers of regions for each sample. The
exact same number of regions may not be selected for each sample because
sometimes multiple genomic regions may have the same ‘score’ value. In
this case, all of the tied genomic regions are retained.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_center_expand</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   sample_name    num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control_rep1            10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control_rep2             9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control_rep3             7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment_rep1           9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment_rep2           9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment_rep3           8</span></span>
<span></span>
<span><span class="fu"><a href="../reference/filter_regions.html">filter_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_center_expand</span>,</span>
<span>  include_top_n_scoring <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">sample_name</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>num_regions <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 2</span></span></span>
<span><span class="co">#&gt;   sample_name    num_regions</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> control_rep1             8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> control_rep2             8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> control_rep3             7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> treatment_rep1           8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> treatment_rep2           8</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> treatment_rep3           8</span></span></code></pre></div>
<p>We requested that only the top 8 genomic regions would be retained
for each sample, and we can see in the information messages that <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
that one sample (‘control03’) contains less then the required 8 sites.
For the remaining samples, we select the expected 8 regions per
sample.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="combine-regions">Combine regions<a class="anchor" aria-label="anchor" href="#combine-regions"></a>
</h2>
<p>After loading and preparing the regions from your samples, you can
use them to build your set of consensus regions using
<code>combine_regions</code>.</p>
<p>The most important parameter that affects the final set of consensus
regions is ‘found_in_samples’, which allows you to retain a genomic
region based on how many samples it was found in (counting the unique
entries in ‘sample_name’). It is up to you to define this parameter
based on your input data and analysis goals. The default is ‘2’ samples,
as we assume that each condition has three biological replicates and
most users would like to focus on regions to be found in at lest two of
these replicates. See the explained in detail section “<a href="#define-parameter-to-best-combine-regions">Define parameter to
best combine regions</a>” for more considerations on setting this
parameter.</p>
<p>The other parameters allow you to configure how the new consensus
regions are annotated. For example, you can define how the ‘center’
column is populated (‘combined_center’) and what new ‘sample_name’
should be used for the new consensus regions (‘combined_sample_name’).
Based on the selected value for the parameter ‘combined_center’, the new
column ‘score’ is filled (for details see section <a href="#which-center-to-select-for-the-consensus-regions">Which center to
select for the consensus regions?</a>). If you want to trace which
samples and genomic regions contributed to a new consensus region, you
can use ‘annotate_with_input_names’ to create a new column ‘input_name’
describing links to the input genomic regions.</p>
<div class="section level3">
<h3 id="quick-start-3">Quick start<a class="anchor" aria-label="anchor" href="#quick-start-3"></a>
</h3>
<p>Following is a quick example that uses default parameters.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  combined_sample_name <span class="op">=</span> <span class="st">"my_new_sample_name"</span>,</span>
<span>  show_message <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name                  score strand center sample_name     </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     150   950 my_new_sample_name|1    100 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">2</span>450  <span style="text-decoration: underline;">3</span>150 my_new_sample_name|2     99 .        <span style="text-decoration: underline;">2</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 my_new_sample_name|3     95 .        <span style="text-decoration: underline;">3</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>450  <span style="text-decoration: underline;">5</span>250 my_new_sample_name|4     98 .        <span style="text-decoration: underline;">4</span>800 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>750  <span style="text-decoration: underline;">6</span>450 my_new_sample_name|5     30 .        <span style="text-decoration: underline;">6</span>100 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 my_new_sample_name|6     98 .        <span style="text-decoration: underline;">7</span>000 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2     150   950 my_new_sample_name|7     50 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    150   950 my_new_sample_name|8     95 .         500 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 my_new_sample_name|9     35 .         600 my_new_sample_n…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   850 my_new_sample_name|10    30 .         500 my_new_sample_n…</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-to-combine-regions">Run to combine regions<a class="anchor" aria-label="anchor" href="#run-to-combine-regions"></a>
</h3>
<p>We run the function <code>combine_regions</code> and give the value
‘consensus’ to the parameter <code>combined_sample_name</code> and we
want to have the link to our input regions by setting
<code>annotate_with_input_name</code> to ‘TRUE’.</p>
<p>In brief, a description what the parameter you can define do. For
more details please see below section “<a href="#define-parameter-to-best-combine-regions">Define parameter to
best combine regions</a>”.</p>
<ul>
<li>
<code>found_in_samples</code> - Defines in how many input samples a
genomic site has to be found to be included into the consensus set.</li>
<li>
<code>combined_center</code> - To stay within the expected data
structure of <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>,
we add new values for ‘center’ and ‘score’ at the end of this function
to make it an accepted input for <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
and <code>filter_regions</code>. You can choose between the options
‘nearest’, ‘strongest’ or ‘middle’ for
<code>combined_center</code>.</li>
<li>
<code>annotate_with_input_names</code> - You can define if for each
consensus region the ‘name’ values of all contributing input regions are
combined and saved in a new column ‘input_names’. This allows you to
link back each consensus region to the input regions.</li>
<li>
<code>combined_sample_name</code> - You can define the new value for
the column ‘sample_name’, which is then used to create the unique ‘name’
column.</li>
</ul>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">consensus_peak_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  found_in_samples <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  combined_center <span class="op">=</span> <span class="st">"nearest"</span>,</span>
<span>  annotate_with_input_names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  combined_sample_name <span class="op">=</span> <span class="st">"consensus"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; → Checking <span style="color: #0000BB;">&lt;class&gt;</span> and <span style="color: #0000BB;">"values"</span> of all columns.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Structure of data was successfully checked to be an accepted input.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Start with the disjoining and filtering genomic regions.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Disjoin and filter by `found_in_samples` of genomic regions successfully</span></span>
<span><span class="co">#&gt;   finished.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Start with combining remaining genomic regions.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Combining remaining genomic regions was successfully finished.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Start with identification of overlaps between the original summit and</span></span>
<span><span class="co">#&gt;   remaining genomic regions.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Remaining regions without overlap will be removed.</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(ranking)`</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Retained genomic regions with input data summit overlap was successfully</span></span>
<span><span class="co">#&gt;   finished.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → Information from input <span style="color: #00BB00;">center</span> will be added to output data frame.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Argument `combined_center` was defined as <span style="color: #0000BB;">"nearest"</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The mean of all input centers is calculated and the nearest input <span style="color: #00BB00;">center</span> is</span></span>
<span><span class="co">#&gt;   used</span></span>
<span><span class="co">#&gt; → Center information in <span style="color: #00BB00;">center</span> and <span style="color: #00BB00;">score</span> are added to the output data frame.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Output data frame columns <span style="color: #00BB00;">center</span> and <span style="color: #00BB00;">score</span> were successfully populated.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; → The value <span style="color: #0000BB;">"consensus"</span> for column <span style="color: #00BB00;">sample_name</span> was provided.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Column <span style="color: #00BB00;">sample_name</span> is filled with provided value consensus.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Column <span style="color: #00BB00;">name</span> is created as unique identifier for each row containing</span></span>
<span><span class="co">#&gt;   <span style="color: #00BB00;">sample_name</span> and the row number.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> The columns <span style="color: #00BB00;">sample_name</span> and <span style="color: #00BB00;">name</span> were successfully populated.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Argument `annotate_with_input_names` was set to <span style="color: #0000BB;">"TRUE"</span>.</span></span>
<span><span class="co">#&gt; → Column <span style="color: #00BB00;">input_names</span> is added to output data frame.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Additional column <span style="color: #00BB00;">input_names</span> was successfully populated.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> All required columns in output data frame were successfully populated.</span></span>
<span><span class="co">#&gt;   </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Genomic regions were successfully combined.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="va">consensus_peak_list</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 9</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name         score strand center sample_name input_names  </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     150   950 consensus|1    100 .         500 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">2</span>450  <span style="text-decoration: underline;">3</span>150 consensus|2     99 .        <span style="text-decoration: underline;">2</span>800 consensus   treatment_re…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 consensus|3     95 .        <span style="text-decoration: underline;">3</span>800 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>450  <span style="text-decoration: underline;">5</span>250 consensus|4     98 .        <span style="text-decoration: underline;">4</span>800 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>750  <span style="text-decoration: underline;">6</span>450 consensus|5     30 .        <span style="text-decoration: underline;">6</span>100 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 consensus|6     98 .        <span style="text-decoration: underline;">7</span>000 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2     150   950 consensus|7     50 .         500 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    150   950 consensus|8     95 .         500 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 consensus|9     35 .         600 consensus   control_rep1…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   850 consensus|10    30 .         500 consensus   control_rep1…</span></span></code></pre></div>
<p>Here you can see the resulting consensus regions tibble with all the
known column and the newly added ‘input_names’. In conclusion, we
provide here multiple options for you to customize the resulting regions
file based on your personal needs.</p>
</div>
<div class="section level3">
<h3 id="explained-in-detail-2">Explained in detail<a class="anchor" aria-label="anchor" href="#explained-in-detail-2"></a>
</h3>
<p>Here we briefly describe what is happening under the hood.</p>
<ul>
<li>
<p><strong>Binning and counting of each genomic input
region:</strong></p>
<p>First step is to apply the GenomicRanges function
<code>disjoin</code> to split up peaks depending on the number of input
samples contributing. These numbers are counted to get the coverage of
each genomic region. These counts are then used for filtering using the
parameter <code>found_in_samples</code>, which is defined by the user.
If <code>found_in_samples</code> is a fraction between 0 and 1, then
only include genomic regions that are found in at least this fraction of
input samples. Default value is 2. <em>Please note</em> if
<code>found_in_samples</code> is set to 1 all genomic regions covered in
at least one sample are included, this makes this function to simply
merge all input regions.</p>
</li>
<li>
<p><strong>Recombine separated genomic regions:</strong></p>
<p>As next step we perform <code>reduce</code>, another GRanges
operation, to combine regions adjacent to each other. If regions are
separated by at least a single nucelotide, we keep them
separately.</p>
</li>
<li>
<p><strong>Overlap with summits:</strong></p>
<p>This is a quality control function for the resulting genomic regions
specifically developed for <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>.
To remove potential false positive sites, we designed the function
<code>combine_regions</code> to check for an overlap between newly
combined regions and input summits. If at least one overlap is found,
the new region will be maintained, otherwise it is removed.</p>
</li>
<li>
<p><strong>Restore data structure</strong>:</p>
<p>In this final step, we restore the accepted data structure for
working with <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
functions to ensure the modularity of the package. This allows a user to
apply the functions <code>center_expand_resgions</code> and
<code>filter_resgions</code> to the combined consensus region file.
Therefore, ‘center’, ‘score’ and ‘sample_name’ columns are created and
populated based on the user defined parameters.</p>
</li>
</ul>
<div class="section level4">
<h4 id="define-parameter-to-best-combine-regions">Define parameter to best combine regions<a class="anchor" aria-label="anchor" href="#define-parameter-to-best-combine-regions"></a>
</h4>
<p>Here we explain the parameters you can set for the function
<code>combine_regions</code> and the defaults we recommend.</p>
<div class="section level5">
<h5 id="in-how-many-samples-should-a-region-be-found">In how many samples should a region be found?<a class="anchor" aria-label="anchor" href="#in-how-many-samples-should-a-region-be-found"></a>
</h5>
<p>As described in “<a href="#run-to-combine-regions">Run to combine
regions</a>”, the default behavior is to retain genomic regions that
overlap in at least two samples. When should you considering using a
non-default value?</p>
<p>Setting <code>found_in_samples</code> = 1 includes every region from
every sample. This is useful if you simply want a union of all peak
regions regardless of how often they were found across your replicates.
For example, you may already be very confident in the quality of the
input genomic regions or you have previously run <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
on subsets of your data to get consensus regions for replicates of the
same condition.</p>
<p>When you have many replicates per condition, you may find it helpful
to increase the stringency by increasing <code>found_in_samples</code>
parameter value.</p>
<p>In general, as you increase <code>found_in_samples</code>, <em><a href="https://bioconductor.org/packages/3.21/peakCombiner" class="external-link">peakCombiner</a></em>
will return fewer and smaller consensus genomic regions.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  found_in_samples <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  combined_sample_name <span class="op">=</span> <span class="st">"found_in_samples_2_example"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name                       score strand center sample_name</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1     150   950 found_in_samples_2_exampl…   100 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">2</span>450  <span style="text-decoration: underline;">3</span>150 found_in_samples_2_exampl…    99 .        <span style="text-decoration: underline;">2</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>150 found_in_samples_2_exampl…    95 .        <span style="text-decoration: underline;">3</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">4</span>450  <span style="text-decoration: underline;">5</span>250 found_in_samples_2_exampl…    98 .        <span style="text-decoration: underline;">4</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">5</span>750  <span style="text-decoration: underline;">6</span>450 found_in_samples_2_exampl…    30 .        <span style="text-decoration: underline;">6</span>100 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 found_in_samples_2_exampl…    98 .        <span style="text-decoration: underline;">7</span>000 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2     150   950 found_in_samples_2_exampl…    50 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr10    150   950 found_in_samples_2_exampl…    95 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4 2   250   950 found_in_samples_2_exampl…    35 .         600 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4-2   250   850 found_in_samples_2_exampl…    30 .         500 found_in_s…</span></span></code></pre></div>
<p>If the parameter <code>found_in_samples</code> is set to ‘1’, this
function basically merges all input regions.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  found_in_samples <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  combined_sample_name <span class="op">=</span> <span class="st">"found_in_samples_1_example"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 8</span></span></span>
<span><span class="co">#&gt;    chrom  start   end name                       score strand center sample_name</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1       1   950 found_in_samples_1_exampl…   100 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1    <span style="text-decoration: underline;">1</span>050  <span style="text-decoration: underline;">2</span>350 found_in_samples_1_exampl…    98 .        <span style="text-decoration: underline;">2</span>000 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1    <span style="text-decoration: underline;">2</span>450  <span style="text-decoration: underline;">3</span>350 found_in_samples_1_exampl…    99 .        <span style="text-decoration: underline;">2</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1    <span style="text-decoration: underline;">3</span>450  <span style="text-decoration: underline;">4</span>250 found_in_samples_1_exampl…    95 .        <span style="text-decoration: underline;">3</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1    <span style="text-decoration: underline;">4</span>350  <span style="text-decoration: underline;">5</span>250 found_in_samples_1_exampl…    98 .        <span style="text-decoration: underline;">4</span>800 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1    <span style="text-decoration: underline;">5</span>350  <span style="text-decoration: underline;">6</span>550 found_in_samples_1_exampl…    30 .        <span style="text-decoration: underline;">6</span>100 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1    <span style="text-decoration: underline;">6</span>650  <span style="text-decoration: underline;">7</span>350 found_in_samples_1_exampl…    98 .        <span style="text-decoration: underline;">7</span>000 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2      50   950 found_in_samples_1_exampl…    50 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr10     50   950 found_in_samples_1_exampl…    95 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr42    250   950 found_in_samples_1_exampl…    90 .         600 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> chr4 2    50   950 found_in_samples_1_exampl…    35 .         600 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> chr4-2   150   950 found_in_samples_1_exampl…    30 .         500 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> chr4?2    50   750 found_in_samples_1_exampl…    25 .         400 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> chr4|2    50   750 found_in_samples_1_exampl…    80 .         400 found_in_s…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> Chr2     350  <span style="text-decoration: underline;">1</span>050 found_in_samples_1_exampl…    80 .         700 found_in_s…</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="which-center-to-select-for-the-consensus-regions">Which center to select for the consensus regions?<a class="anchor" aria-label="anchor" href="#which-center-to-select-for-the-consensus-regions"></a>
</h5>
<p>You only need to consider how to select the center of your new
genomic region if you plan to run <a href="#center-and-expand-regions"><code>center_expand_regions</code></a>
to update all the consensus genomic regions so that they are the same
size, or if you plan to use the ‘score’ value for your downstream
analyses. Each consensus region can be derived from several input
regions, each of which has its own ‘center’ and ‘score’ value.</p>
<p>There are three ways you can define the ‘center’ of new genomic
regions. Your choice also defines how the ‘score’ is calculated.</p>
<p>Three different option are available:</p>
<ul>
<li>
<code>middle</code> - the mathematical midpoint of the new region.
The resulting ‘score’ for each consensus region is the mean of all input
score values contributing to that site.</li>
<li>
<code>strongest</code> - the ‘center’ of the input region that has
the the highest ‘score’ of all overlapping input regions. ‘score’ is
taken from the used input center.</li>
<li>
<code>nearest</code> - the ‘center’ of the input region that is
closest to mean of the ‘center’s of all overlapping input regions
(default). ’score’ is taken from the used input center.</li>
</ul>
<p>Lets have a look at the ‘center’ and ‘score’ within the synthetic
data, when setting <code>combined_center</code> to ‘strongest’.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  combined_center <span class="op">=</span> <span class="st">"strongest"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"center"</span>, <span class="st">"score"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span></span>
<span><span class="co">#&gt;    center score</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    500   100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   <span style="text-decoration: underline;">2</span>800    99</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   <span style="text-decoration: underline;">3</span>800    95</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   <span style="text-decoration: underline;">4</span>800    98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   <span style="text-decoration: underline;">6</span>200    98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   <span style="text-decoration: underline;">7</span>000    98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>    600    50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>    400    95</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>    600    35</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    500    30</span></span></code></pre></div>
<p>Or to ‘middle’.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  combined_center <span class="op">=</span> <span class="st">"middle"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"center"</span>, <span class="st">"score"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span></span>
<span><span class="co">#&gt;    center score</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    550  96.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   <span style="text-decoration: underline;">2</span>800  98  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   <span style="text-decoration: underline;">3</span>800  60.7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   <span style="text-decoration: underline;">4</span>850  81.3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   <span style="text-decoration: underline;">6</span>100  51.3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   <span style="text-decoration: underline;">7</span>000  56  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>    550  30  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>    550  87.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>    600  30  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    550  25</span></span></code></pre></div>
<p>Or to ‘nearest’.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/combine_regions.html">combine_regions</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">data_filtered</span>,</span>
<span>  combined_center <span class="op">=</span> <span class="st">"nearest"</span>,</span>
<span>  show_messages <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"center"</span>, <span class="st">"score"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span></span>
<span><span class="co">#&gt;    center score</span></span>
<span><span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>    500   100</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>   <span style="text-decoration: underline;">2</span>800    99</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>   <span style="text-decoration: underline;">3</span>800    95</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>   <span style="text-decoration: underline;">4</span>800    98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>   <span style="text-decoration: underline;">6</span>100    30</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>   <span style="text-decoration: underline;">7</span>000    98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>    500    50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>    500    95</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>    600    35</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span>    500    30</span></span></code></pre></div>
<p>We recommend to use the <code>nearest</code> option as this will give
you a best common center and associated score.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>This vignette was built using:</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.7.2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] peakCombiner_0.99.1 testthat_3.2.3      BiocStyle_2.35.0   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6            ggplot2_3.5.1           xfun_0.50              </span></span>
<span><span class="co">#&gt;  [4] bslib_0.9.0             htmlwidgets_1.6.4       devtools_2.4.5         </span></span>
<span><span class="co">#&gt;  [7] remotes_2.5.0           tzdb_0.4.0              vctrs_0.6.5            </span></span>
<span><span class="co">#&gt; [10] tools_4.4.2             generics_0.1.3          stats4_4.4.2           </span></span>
<span><span class="co">#&gt; [13] tibble_3.2.1            pkgconfig_2.0.3         desc_1.4.3             </span></span>
<span><span class="co">#&gt; [16] S4Vectors_0.45.4        lifecycle_1.0.4         GenomeInfoDbData_1.2.13</span></span>
<span><span class="co">#&gt; [19] farver_2.1.2            compiler_4.4.2          stringr_1.5.1          </span></span>
<span><span class="co">#&gt; [22] brio_1.1.5              textshaping_1.0.0       munsell_0.5.1          </span></span>
<span><span class="co">#&gt; [25] httpuv_1.6.15           GenomeInfoDb_1.43.4     htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [28] usethis_3.1.0           sass_0.4.9              yaml_2.3.10            </span></span>
<span><span class="co">#&gt; [31] later_1.4.1             pillar_1.10.1           pkgdown_2.1.1.9000     </span></span>
<span><span class="co">#&gt; [34] jquerylib_0.1.4         urlchecker_1.0.1        tidyr_1.3.1            </span></span>
<span><span class="co">#&gt; [37] ellipsis_0.3.2          cachem_1.1.0            sessioninfo_1.2.3      </span></span>
<span><span class="co">#&gt; [40] mime_0.12               tidyselect_1.2.1        digest_0.6.37          </span></span>
<span><span class="co">#&gt; [43] stringi_1.8.4           dplyr_1.1.4             purrr_1.0.4            </span></span>
<span><span class="co">#&gt; [46] bookdown_0.42           labeling_0.4.3          grid_4.4.2             </span></span>
<span><span class="co">#&gt; [49] rprojroot_2.0.4         fastmap_1.2.0           colorspace_2.1-1       </span></span>
<span><span class="co">#&gt; [52] here_1.0.1              cli_3.6.4               magrittr_2.0.3         </span></span>
<span><span class="co">#&gt; [55] utf8_1.2.4              pkgbuild_1.4.6          readr_2.1.5            </span></span>
<span><span class="co">#&gt; [58] withr_3.0.2             scales_1.3.0            UCSC.utils_1.3.1       </span></span>
<span><span class="co">#&gt; [61] promises_1.3.2          rmarkdown_2.29          XVector_0.47.2         </span></span>
<span><span class="co">#&gt; [64] httr_1.4.7              qpdf_1.3.4              askpass_1.2.1          </span></span>
<span><span class="co">#&gt; [67] ragg_1.3.3              hms_1.1.3               memoise_2.0.1          </span></span>
<span><span class="co">#&gt; [70] shiny_1.10.0            evaluate_1.0.3          knitr_1.49             </span></span>
<span><span class="co">#&gt; [73] GenomicRanges_1.59.1    IRanges_2.41.3          miniUI_0.1.1.1         </span></span>
<span><span class="co">#&gt; [76] profvis_0.4.0           rlang_1.1.5             Rcpp_1.0.14            </span></span>
<span><span class="co">#&gt; [79] xtable_1.8-4            glue_1.8.0              BiocManager_1.30.25    </span></span>
<span><span class="co">#&gt; [82] BiocGenerics_0.53.6     pkgload_1.4.0           rstudioapi_0.17.1      </span></span>
<span><span class="co">#&gt; [85] jsonlite_1.8.9          R6_2.6.0                systemfonts_1.2.1      </span></span>
<span><span class="co">#&gt; [88] fs_1.6.5</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-macs" class="csl-entry">
Zhang, Yong, Tao Liu, Clifford A Meyer, Jérôme Eeckhoute, David S
Johnson, Bradley E Bernstein, Chad Nusbaum, et al. 2008.
<span>“Model-Based Analysis of ChIP-Seq (<span>MACS</span>).”</span>
<em>Genome Biology</em> 9 (9): R137. <a href="https://doi.org/doi:10.1186/gb-2008-9-9-r137" class="external-link">https://doi.org/doi:10.1186/gb-2008-9-9-r137</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Markus Muckenhuber, Kathleen Sprouffske, Michael Stadler.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.9000.</p>
</div>

      </footer>
</div>






  </body>
</html>
